// node-appwrite/dist/client.mjs
import { Agent, FormData, File, fetch } from 'undici';
import { InputFile } from './inputFile.mjs';
import JSONbigModule from 'json-bigint';
export { Query } from './query.mjs';

// src/client.ts
var JSONbigParser = JSONbigModule({ storeAsString: false });
var JSONbigSerializer = JSONbigModule({ useNativeBigInt: true });
var MAX_SAFE = BigInt(Number.MAX_SAFE_INTEGER);
var MIN_SAFE = BigInt(Number.MIN_SAFE_INTEGER);
var MAX_INT64 = BigInt("9223372036854775807");
var MIN_INT64 = BigInt("-9223372036854775808");
function isBigNumber(value) {
  return value !== null && typeof value === "object" && value._isBigNumber === true && typeof value.isInteger === "function" && typeof value.toFixed === "function" && typeof value.toNumber === "function";
}
function reviver(_key, value) {
  if (isBigNumber(value)) {
    if (value.isInteger()) {
      const str = value.toFixed();
      const bi = BigInt(str);
      if (bi >= MIN_SAFE && bi <= MAX_SAFE) {
        return Number(str);
      }
      if (bi >= MIN_INT64 && bi <= MAX_INT64) {
        return bi;
      }
      return value.toNumber();
    }
    return value.toNumber();
  }
  return value;
}
var JSONbig = {
  parse: (text) => JSONbigParser.parse(text, reviver),
  stringify: JSONbigSerializer.stringify
};
var AppwriteException = class extends Error {
  constructor(message, code = 0, type = "", response = "") {
    super(message);
    this.name = "AppwriteException";
    this.message = message;
    this.code = code;
    this.type = type;
    this.response = response;
  }
};
function getUserAgent() {
  let ua = "AppwriteNodeJSSDK/27.0.0";
  const platform = [];
  if (typeof process !== "undefined") {
    if (typeof process.platform === "string") platform.push(process.platform);
    if (typeof process.arch === "string") platform.push(process.arch);
  }
  if (platform.length > 0) {
    ua += ` (${platform.join("; ")})`;
  }
  if (typeof navigator !== "undefined" && typeof navigator.userAgent === "string") {
    ua += ` ${navigator.userAgent}`;
  } else if (typeof globalThis.EdgeRuntime === "string") {
    ua += ` EdgeRuntime`;
  } else if (typeof process !== "undefined" && typeof process.version === "string") {
    ua += ` Node.js/${process.version}`;
  }
  return ua;
}
var _Client = class _Client {
  constructor() {
    this.config = {
      endpoint: "https://cloud.appwrite.io/v1",
      selfSigned: false,
      project: "",
      key: "",
      jwt: "",
      bearer: "",
      locale: "",
      session: "",
      forwardeduseragent: "",
      devkey: "",
      cookie: "",
      impersonateuserid: "",
      impersonateuseremail: "",
      impersonateuserphone: ""
    };
    this.headers = {
      "x-sdk-name": "Node.js",
      "x-sdk-platform": "server",
      "x-sdk-language": "nodejs",
      "x-sdk-version": "27.0.0",
      "user-agent": getUserAgent(),
      "X-Appwrite-Response-Format": "1.9.5"
    };
  }
  /**
   * Set Endpoint
   *
   * Your project endpoint
   *
   * @param {string} endpoint
   *
   * @returns {this}
   */
  setEndpoint(endpoint) {
    if (!endpoint || typeof endpoint !== "string") {
      throw new AppwriteException("Endpoint must be a valid string");
    }
    if (!endpoint.startsWith("http://") && !endpoint.startsWith("https://")) {
      throw new AppwriteException("Invalid endpoint URL: " + endpoint);
    }
    this.config.endpoint = endpoint;
    return this;
  }
  /**
   * Set self-signed
   *
   * @param {boolean} selfSigned
   *
   * @returns {this}
   */
  setSelfSigned(selfSigned) {
    if (typeof globalThis.EdgeRuntime !== "undefined") {
      console.warn("setSelfSigned is not supported in edge runtimes.");
    }
    this.config.selfSigned = selfSigned;
    if (!selfSigned && this.selfSignedAgent) {
      this.selfSignedAgent.destroy();
      this.selfSignedAgent = void 0;
    }
    return this;
  }
  /**
   * Add header
   *
   * @param {string} header
   * @param {string} value
   *
   * @returns {this}
   */
  addHeader(header, value) {
    this.headers[header.toLowerCase()] = value;
    return this;
  }
  /**
   * Get Headers
   *
   * Returns a copy of the current request headers, including any
   * authentication headers. Handle with care.
   *
   * @returns {Headers}
   */
  getHeaders() {
    return { ...this.headers };
  }
  /**
   * Set Project
   *
   * Your project ID
   *
   * @param value string
   *
   * @return {this}
   */
  setProject(value) {
    this.config.project = value;
    return this;
  }
  /**
   * Set Key
   *
   * Your secret API key
   *
   * @param value string
   *
   * @return {this}
   */
  setKey(value) {
    this.headers["X-Appwrite-Key"] = value;
    this.config.key = value;
    return this;
  }
  /**
   * Set JWT
   *
   * Your secret JSON Web Token
   *
   * @param value string
   *
   * @return {this}
   */
  setJWT(value) {
    this.headers["X-Appwrite-JWT"] = value;
    this.config.jwt = value;
    return this;
  }
  /**
   * Set Bearer
   *
   * The OAuth access token to authenticate with
   *
   * @param value string
   *
   * @return {this}
   */
  setBearer(value) {
    this.headers["Authorization"] = `Bearer ${value}`;
    this.config.bearer = value;
    return this;
  }
  /**
   * Set Locale
   *
   * @param value string
   *
   * @return {this}
   */
  setLocale(value) {
    this.headers["X-Appwrite-Locale"] = value;
    this.config.locale = value;
    return this;
  }
  /**
   * Set Session
   *
   * The user session to authenticate with
   *
   * @param value string
   *
   * @return {this}
   */
  setSession(value) {
    this.headers["X-Appwrite-Session"] = value;
    this.config.session = value;
    return this;
  }
  /**
   * Set ForwardedUserAgent
   *
   * The user agent string of the client that made the request
   *
   * @param value string
   *
   * @return {this}
   */
  setForwardedUserAgent(value) {
    this.headers["X-Forwarded-User-Agent"] = value;
    this.config.forwardeduseragent = value;
    return this;
  }
  /**
   * Set DevKey
   *
   * Your secret dev API key
   *
   * @param value string
   *
   * @return {this}
   */
  setDevKey(value) {
    this.headers["X-Appwrite-Dev-Key"] = value;
    this.config.devkey = value;
    return this;
  }
  /**
   * Set Cookie
   *
   * The user cookie to authenticate with. Used by SDKs that forward an incoming Cookie header in server-side runtimes.
   *
   * @param value string
   *
   * @return {this}
   */
  setCookie(value) {
    this.headers["Cookie"] = value;
    this.config.cookie = value;
    return this;
  }
  /**
   * Set ImpersonateUserId
   *
   * Impersonate a user by ID
   *
   * @param value string
   *
   * @return {this}
   */
  setImpersonateUserId(value) {
    this.headers["X-Appwrite-Impersonate-User-Id"] = value;
    this.config.impersonateuserid = value;
    return this;
  }
  /**
   * Set ImpersonateUserEmail
   *
   * Impersonate a user by email
   *
   * @param value string
   *
   * @return {this}
   */
  setImpersonateUserEmail(value) {
    this.headers["X-Appwrite-Impersonate-User-Email"] = value;
    this.config.impersonateuseremail = value;
    return this;
  }
  /**
   * Set ImpersonateUserPhone
   *
   * Impersonate a user by phone
   *
   * @param value string
   *
   * @return {this}
   */
  setImpersonateUserPhone(value) {
    this.headers["X-Appwrite-Impersonate-User-Phone"] = value;
    this.config.impersonateuserphone = value;
    return this;
  }
  prepareRequest(method, url, headers = {}, params = {}) {
    method = method.toUpperCase();
    headers = Object.assign({}, this.headers, headers);
    let options = {
      method,
      headers
    };
    if (this.config.selfSigned) {
      if (!this.selfSignedAgent) {
        this.selfSignedAgent = new Agent({
          connect: {
            rejectUnauthorized: false
          }
        });
      }
      options.dispatcher = this.selfSignedAgent;
    }
    if (method === "GET") {
      for (const [key, value] of Object.entries(_Client.flatten(params))) {
        url.searchParams.append(key, value);
      }
    } else {
      switch (headers["content-type"]) {
        case "application/json":
          options.body = JSONbig.stringify(params);
          break;
        case "multipart/form-data":
          const formData = new FormData();
          for (const [key, value] of Object.entries(params)) {
            if (value instanceof File) {
              formData.append(key, value, value.name);
            } else if (Array.isArray(value)) {
              for (const nestedValue of value) {
                formData.append(`${key}[]`, nestedValue);
              }
            } else {
              formData.append(key, value);
            }
          }
          options.body = formData;
          delete headers["content-type"];
          break;
      }
    }
    return { uri: url.toString(), options };
  }
  async chunkedUpload(method, url, headers = {}, originalPayload = {}, onProgress) {
    const [fileParam, file] = Object.entries(originalPayload).find(
      ([_, value]) => value instanceof File || value instanceof InputFile
    ) ?? [];
    if (!file || !fileParam) {
      throw new Error("File not found in payload");
    }
    if (file instanceof InputFile) {
      const size = await file.size();
      if (size <= _Client.CHUNK_SIZE) {
        const payload = { ...originalPayload };
        payload[fileParam] = await file.toFile();
        return await this.call(method, url, headers, payload);
      }
      const totalChunks2 = Math.ceil(size / _Client.CHUNK_SIZE);
      const firstChunkEnd2 = Math.min(_Client.CHUNK_SIZE, size);
      const firstChunkHeaders2 = { ...headers, "content-range": `bytes 0-${firstChunkEnd2 - 1}/${size}` };
      const firstChunk2 = await file.slice(0, firstChunkEnd2);
      const firstPayload2 = { ...originalPayload };
      firstPayload2[fileParam] = new File([firstChunk2], file.filename);
      let response2 = await this.call(method, url, firstChunkHeaders2, firstPayload2);
      const uploadId2 = response2 == null ? void 0 : response2.$id;
      if (onProgress && typeof onProgress === "function") {
        onProgress({
          $id: uploadId2,
          progress: Math.round(firstChunkEnd2 / size * 100),
          sizeUploaded: firstChunkEnd2,
          chunksTotal: totalChunks2,
          chunksUploaded: 1
        });
      }
      if (totalChunks2 === 1) {
        return response2;
      }
      const chunks2 = [];
      for (let i = 1; i < totalChunks2; i++) {
        const start = i * _Client.CHUNK_SIZE;
        const end = Math.min(start + _Client.CHUNK_SIZE, size);
        chunks2.push({ start, end });
      }
      const CONCURRENCY2 = 8;
      let completedCount2 = 1;
      let uploadedBytes2 = firstChunkEnd2;
      let lastResponse2 = response2;
      let finalResponse2 = null;
      let failed2 = false;
      const isUploadComplete2 = (chunkResponse) => {
        const chunksUploaded = chunkResponse == null ? void 0 : chunkResponse.chunksUploaded;
        const chunksTotal = (chunkResponse == null ? void 0 : chunkResponse.chunksTotal) ?? totalChunks2;
        return typeof chunksUploaded === "number" && typeof chunksTotal === "number" && chunksUploaded >= chunksTotal;
      };
      const uploadChunk2 = async (chunk) => {
        const chunkHeaders = { ...headers };
        if (uploadId2) {
          chunkHeaders["x-appwrite-id"] = uploadId2;
        }
        chunkHeaders["content-range"] = `bytes ${chunk.start}-${chunk.end - 1}/${size}`;
        const chunkBlob = await file.slice(chunk.start, chunk.end);
        const chunkPayload = { ...originalPayload };
        chunkPayload[fileParam] = new File([chunkBlob], file.filename);
        const chunkResponse = await this.call(method, url, chunkHeaders, chunkPayload);
        if (failed2) {
          return chunkResponse;
        }
        completedCount2++;
        uploadedBytes2 += chunk.end - chunk.start;
        lastResponse2 = chunkResponse;
        if (isUploadComplete2(chunkResponse)) {
          finalResponse2 = chunkResponse;
        }
        if (onProgress && typeof onProgress === "function") {
          onProgress({
            $id: uploadId2,
            progress: Math.round(uploadedBytes2 / size * 100),
            sizeUploaded: uploadedBytes2,
            chunksTotal: totalChunks2,
            chunksUploaded: completedCount2
          });
        }
        return chunkResponse;
      };
      const queue2 = [...chunks2];
      const workers2 = [];
      const workerCount2 = Math.min(CONCURRENCY2, queue2.length);
      for (let i = 0; i < workerCount2; i++) {
        workers2.push(
          (async () => {
            while (!failed2 && queue2.length > 0) {
              const chunk = queue2.shift();
              try {
                await uploadChunk2(chunk);
              } catch (error) {
                failed2 = true;
                throw error;
              }
            }
          })()
        );
      }
      await Promise.all(workers2);
      return finalResponse2 ?? lastResponse2;
    }
    if (file.size <= _Client.CHUNK_SIZE) {
      return await this.call(method, url, headers, originalPayload);
    }
    const totalChunks = Math.ceil(file.size / _Client.CHUNK_SIZE);
    const firstChunkEnd = Math.min(_Client.CHUNK_SIZE, file.size);
    const firstChunkHeaders = { ...headers, "content-range": `bytes 0-${firstChunkEnd - 1}/${file.size}` };
    const firstChunk = file.slice(0, firstChunkEnd);
    const firstPayload = { ...originalPayload };
    firstPayload[fileParam] = new File([firstChunk], file.name);
    let response = await this.call(method, url, firstChunkHeaders, firstPayload);
    const uploadId = response == null ? void 0 : response.$id;
    if (onProgress && typeof onProgress === "function") {
      onProgress({
        $id: uploadId,
        progress: Math.round(firstChunkEnd / file.size * 100),
        sizeUploaded: firstChunkEnd,
        chunksTotal: totalChunks,
        chunksUploaded: 1
      });
    }
    if (totalChunks === 1) {
      return response;
    }
    const chunks = [];
    for (let i = 1; i < totalChunks; i++) {
      const start = i * _Client.CHUNK_SIZE;
      const end = Math.min(start + _Client.CHUNK_SIZE, file.size);
      chunks.push({ start, end });
    }
    const CONCURRENCY = 8;
    let completedCount = 1;
    let uploadedBytes = firstChunkEnd;
    let lastResponse = response;
    let finalResponse = null;
    let failed = false;
    const isUploadComplete = (chunkResponse) => {
      const chunksUploaded = chunkResponse == null ? void 0 : chunkResponse.chunksUploaded;
      const chunksTotal = (chunkResponse == null ? void 0 : chunkResponse.chunksTotal) ?? totalChunks;
      return typeof chunksUploaded === "number" && typeof chunksTotal === "number" && chunksUploaded >= chunksTotal;
    };
    const uploadChunk = async (chunk) => {
      const chunkHeaders = { ...headers };
      if (uploadId) {
        chunkHeaders["x-appwrite-id"] = uploadId;
      }
      chunkHeaders["content-range"] = `bytes ${chunk.start}-${chunk.end - 1}/${file.size}`;
      const chunkBlob = file.slice(chunk.start, chunk.end);
      const chunkPayload = { ...originalPayload };
      chunkPayload[fileParam] = new File([chunkBlob], file.name);
      const chunkResponse = await this.call(method, url, chunkHeaders, chunkPayload);
      if (failed) {
        return chunkResponse;
      }
      completedCount++;
      uploadedBytes += chunk.end - chunk.start;
      lastResponse = chunkResponse;
      if (isUploadComplete(chunkResponse)) {
        finalResponse = chunkResponse;
      }
      if (onProgress && typeof onProgress === "function") {
        onProgress({
          $id: uploadId,
          progress: Math.round(uploadedBytes / file.size * 100),
          sizeUploaded: uploadedBytes,
          chunksTotal: totalChunks,
          chunksUploaded: completedCount
        });
      }
      return chunkResponse;
    };
    const queue = [...chunks];
    const workers = [];
    const workerCount = Math.min(CONCURRENCY, queue.length);
    for (let i = 0; i < workerCount; i++) {
      workers.push(
        (async () => {
          while (!failed && queue.length > 0) {
            const chunk = queue.shift();
            try {
              await uploadChunk(chunk);
            } catch (error) {
              failed = true;
              throw error;
            }
          }
        })()
      );
    }
    await Promise.all(workers);
    return finalResponse ?? lastResponse;
  }
  async ping() {
    return this.call("GET", new URL(this.config.endpoint + "/ping"), {
      "X-Appwrite-Project": this.config.project,
      "accept": "application/json"
    });
  }
  async redirect(method, url, headers = {}, params = {}) {
    const { uri, options } = this.prepareRequest(method, url, headers, params);
    const response = await fetch(uri, {
      ...options,
      redirect: "manual"
    });
    if (response.status !== 301 && response.status !== 302) {
      throw new AppwriteException("Invalid redirect", response.status);
    }
    return response.headers.get("location") || "";
  }
  async call(method, url, headers = {}, params = {}, responseType = "json") {
    var _a, _b;
    const { uri, options } = this.prepareRequest(method, url, headers, params);
    let data = null;
    const response = await fetch(uri, options);
    const warnings = response.headers.get("x-appwrite-warning");
    if (warnings) {
      warnings.split(";").forEach((warning) => console.warn("Warning: " + warning));
    }
    if ((_a = response.headers.get("content-type")) == null ? void 0 : _a.includes("application/json")) {
      data = JSONbig.parse(await response.text());
    } else if (responseType === "arrayBuffer") {
      data = await response.arrayBuffer();
    } else {
      data = {
        message: await response.text()
      };
    }
    if (400 <= response.status) {
      let responseText = "";
      if (((_b = response.headers.get("content-type")) == null ? void 0 : _b.includes("application/json")) || responseType === "arrayBuffer") {
        responseText = JSONbig.stringify(data);
      } else {
        responseText = data == null ? void 0 : data.message;
      }
      throw new AppwriteException(data == null ? void 0 : data.message, response.status, data == null ? void 0 : data.type, responseText);
    }
    if (data && typeof data === "object") {
      Object.defineProperty(data, "toString", {
        value: () => JSONbig.stringify(data),
        writable: true,
        enumerable: false,
        configurable: true
      });
    }
    return data;
  }
  static flatten(data, prefix = "") {
    let output = {};
    for (const [key, value] of Object.entries(data)) {
      let finalKey = prefix ? prefix + "[" + key + "]" : key;
      if (Array.isArray(value)) {
        output = { ...output, ..._Client.flatten(value, finalKey) };
      } else {
        output[finalKey] = value;
      }
    }
    return output;
  }
};
_Client.CHUNK_SIZE = 1024 * 1024 * 5;
var Client = _Client;

export { AppwriteException, Client };
