// node-appwrite/dist/index.js
'use strict';

var client = require('./client');
var account = require('./services/account');
var activities = require('./services/activities');
var avatars = require('./services/avatars');
var backups = require('./services/backups');
var databases = require('./services/databases');
var functions = require('./services/functions');
var graphql = require('./services/graphql');
var locale = require('./services/locale');
var messaging = require('./services/messaging');
var organization = require('./services/organization');
var presences = require('./services/presences');
var project = require('./services/project');
var proxy = require('./services/proxy');
var advisor = require('./services/advisor');
var sites = require('./services/sites');
var storage = require('./services/storage');
var tablesDb = require('./services/tables-db');
var teams = require('./services/teams');
var tokens = require('./services/tokens');
var users = require('./services/users');
var webhooks = require('./services/webhooks');
var permission = require('./permission');
var role = require('./role');
var id = require('./id');
var operator = require('./operator');
var authenticatorType = require('./enums/authenticator-type');
var authenticationFactor = require('./enums/authentication-factor');
var oAuthProvider = require('./enums/o-auth-provider');
var browser = require('./enums/browser');
var creditCard = require('./enums/credit-card');
var flag = require('./enums/flag');
var browserTheme = require('./enums/browser-theme');
var timezone = require('./enums/timezone');
var browserPermission = require('./enums/browser-permission');
var imageFormat = require('./enums/image-format');
var backupServices = require('./enums/backup-services');
var relationshipType = require('./enums/relationship-type');
var relationMutate = require('./enums/relation-mutate');
var databasesIndexType = require('./enums/databases-index-type');
var orderBy = require('./enums/order-by');
var runtime = require('./enums/runtime');
var projectKeyScopes = require('./enums/project-key-scopes');
var templateReferenceType = require('./enums/template-reference-type');
var vcsReferenceType = require('./enums/vcs-reference-type');
var deploymentDownloadType = require('./enums/deployment-download-type');
var executionMethod = require('./enums/execution-method');
var messagePriority = require('./enums/message-priority');
var smtpEncryption = require('./enums/smtp-encryption');
var organizationKeyScopes = require('./enums/organization-key-scopes');
var region = require('./enums/region');
var projectAuthMethodId = require('./enums/project-auth-method-id');
var projectOAuth2GooglePrompt = require('./enums/project-o-auth-2-google-prompt');
var projectOAuth2OidcPrompt = require('./enums/project-o-auth-2-oidc-prompt');
var projectOAuthProviderId = require('./enums/project-o-auth-provider-id');
var projectPolicyId = require('./enums/project-policy-id');
var projectProtocolId = require('./enums/project-protocol-id');
var projectServiceId = require('./enums/project-service-id');
var projectSmtpSecure = require('./enums/project-smtp-secure');
var projectEmailTemplateId = require('./enums/project-email-template-id');
var projectEmailTemplateLocale = require('./enums/project-email-template-locale');
var statusCode = require('./enums/status-code');
var proxyResourceType = require('./enums/proxy-resource-type');
var framework = require('./enums/framework');
var buildRuntime = require('./enums/build-runtime');
var adapter = require('./enums/adapter');
var compression = require('./enums/compression');
var imageGravity = require('./enums/image-gravity');
var tablesDbIndexType = require('./enums/tables-db-index-type');
var passwordHash = require('./enums/password-hash');
var messagingProviderType = require('./enums/messaging-provider-type');
var databaseType = require('./enums/database-type');
var databaseStatus = require('./enums/database-status');
var attributeStatus = require('./enums/attribute-status');
var columnStatus = require('./enums/column-status');
var indexStatus = require('./enums/index-status');
var deploymentStatus = require('./enums/deployment-status');
var executionTrigger = require('./enums/execution-trigger');
var executionStatus = require('./enums/execution-status');
var oAuth2GooglePrompt = require('./enums/o-auth-2-google-prompt');
var oAuth2OidcPrompt = require('./enums/o-auth-2-oidc-prompt');
var platformType = require('./enums/platform-type');
var proxyRuleDeploymentResourceType = require('./enums/proxy-rule-deployment-resource-type');
var proxyRuleStatus = require('./enums/proxy-rule-status');
var messageStatus = require('./enums/message-status');
var billingPlanGroup = require('./enums/billing-plan-group');



Object.defineProperty(exports, "AppwriteException", {
  enumerable: true,
  get: function () { return client.AppwriteException; }
});
Object.defineProperty(exports, "Client", {
  enumerable: true,
  get: function () { return client.Client; }
});
Object.defineProperty(exports, "Query", {
  enumerable: true,
  get: function () { return client.Query; }
});
Object.defineProperty(exports, "Account", {
  enumerable: true,
  get: function () { return account.Account; }
});
Object.defineProperty(exports, "Activities", {
  enumerable: true,
  get: function () { return activities.Activities; }
});
Object.defineProperty(exports, "Avatars", {
  enumerable: true,
  get: function () { return avatars.Avatars; }
});
Object.defineProperty(exports, "Backups", {
  enumerable: true,
  get: function () { return backups.Backups; }
});
Object.defineProperty(exports, "Databases", {
  enumerable: true,
  get: function () { return databases.Databases; }
});
Object.defineProperty(exports, "Functions", {
  enumerable: true,
  get: function () { return functions.Functions; }
});
Object.defineProperty(exports, "Graphql", {
  enumerable: true,
  get: function () { return graphql.Graphql; }
});
Object.defineProperty(exports, "Locale", {
  enumerable: true,
  get: function () { return locale.Locale; }
});
Object.defineProperty(exports, "Messaging", {
  enumerable: true,
  get: function () { return messaging.Messaging; }
});
Object.defineProperty(exports, "Organization", {
  enumerable: true,
  get: function () { return organization.Organization; }
});
Object.defineProperty(exports, "Presences", {
  enumerable: true,
  get: function () { return presences.Presences; }
});
Object.defineProperty(exports, "Project", {
  enumerable: true,
  get: function () { return project.Project; }
});
Object.defineProperty(exports, "Proxy", {
  enumerable: true,
  get: function () { return proxy.Proxy; }
});
Object.defineProperty(exports, "Advisor", {
  enumerable: true,
  get: function () { return advisor.Advisor; }
});
Object.defineProperty(exports, "Sites", {
  enumerable: true,
  get: function () { return sites.Sites; }
});
Object.defineProperty(exports, "Storage", {
  enumerable: true,
  get: function () { return storage.Storage; }
});
Object.defineProperty(exports, "TablesDB", {
  enumerable: true,
  get: function () { return tablesDb.TablesDB; }
});
Object.defineProperty(exports, "Teams", {
  enumerable: true,
  get: function () { return teams.Teams; }
});
Object.defineProperty(exports, "Tokens", {
  enumerable: true,
  get: function () { return tokens.Tokens; }
});
Object.defineProperty(exports, "Users", {
  enumerable: true,
  get: function () { return users.Users; }
});
Object.defineProperty(exports, "Webhooks", {
  enumerable: true,
  get: function () { return webhooks.Webhooks; }
});
Object.defineProperty(exports, "Permission", {
  enumerable: true,
  get: function () { return permission.Permission; }
});
Object.defineProperty(exports, "Role", {
  enumerable: true,
  get: function () { return role.Role; }
});
Object.defineProperty(exports, "ID", {
  enumerable: true,
  get: function () { return id.ID; }
});
Object.defineProperty(exports, "Condition", {
  enumerable: true,
  get: function () { return operator.Condition; }
});
Object.defineProperty(exports, "Operator", {
  enumerable: true,
  get: function () { return operator.Operator; }
});
Object.defineProperty(exports, "AuthenticatorType", {
  enumerable: true,
  get: function () { return authenticatorType.AuthenticatorType; }
});
Object.defineProperty(exports, "AuthenticationFactor", {
  enumerable: true,
  get: function () { return authenticationFactor.AuthenticationFactor; }
});
Object.defineProperty(exports, "OAuthProvider", {
  enumerable: true,
  get: function () { return oAuthProvider.OAuthProvider; }
});
Object.defineProperty(exports, "Browser", {
  enumerable: true,
  get: function () { return browser.Browser; }
});
Object.defineProperty(exports, "CreditCard", {
  enumerable: true,
  get: function () { return creditCard.CreditCard; }
});
Object.defineProperty(exports, "Flag", {
  enumerable: true,
  get: function () { return flag.Flag; }
});
Object.defineProperty(exports, "BrowserTheme", {
  enumerable: true,
  get: function () { return browserTheme.BrowserTheme; }
});
Object.defineProperty(exports, "Timezone", {
  enumerable: true,
  get: function () { return timezone.Timezone; }
});
Object.defineProperty(exports, "BrowserPermission", {
  enumerable: true,
  get: function () { return browserPermission.BrowserPermission; }
});
Object.defineProperty(exports, "ImageFormat", {
  enumerable: true,
  get: function () { return imageFormat.ImageFormat; }
});
Object.defineProperty(exports, "BackupServices", {
  enumerable: true,
  get: function () { return backupServices.BackupServices; }
});
Object.defineProperty(exports, "RelationshipType", {
  enumerable: true,
  get: function () { return relationshipType.RelationshipType; }
});
Object.defineProperty(exports, "RelationMutate", {
  enumerable: true,
  get: function () { return relationMutate.RelationMutate; }
});
Object.defineProperty(exports, "DatabasesIndexType", {
  enumerable: true,
  get: function () { return databasesIndexType.DatabasesIndexType; }
});
Object.defineProperty(exports, "OrderBy", {
  enumerable: true,
  get: function () { return orderBy.OrderBy; }
});
Object.defineProperty(exports, "Runtime", {
  enumerable: true,
  get: function () { return runtime.Runtime; }
});
Object.defineProperty(exports, "ProjectKeyScopes", {
  enumerable: true,
  get: function () { return projectKeyScopes.ProjectKeyScopes; }
});
Object.defineProperty(exports, "TemplateReferenceType", {
  enumerable: true,
  get: function () { return templateReferenceType.TemplateReferenceType; }
});
Object.defineProperty(exports, "VCSReferenceType", {
  enumerable: true,
  get: function () { return vcsReferenceType.VCSReferenceType; }
});
Object.defineProperty(exports, "DeploymentDownloadType", {
  enumerable: true,
  get: function () { return deploymentDownloadType.DeploymentDownloadType; }
});
Object.defineProperty(exports, "ExecutionMethod", {
  enumerable: true,
  get: function () { return executionMethod.ExecutionMethod; }
});
Object.defineProperty(exports, "MessagePriority", {
  enumerable: true,
  get: function () { return messagePriority.MessagePriority; }
});
Object.defineProperty(exports, "SmtpEncryption", {
  enumerable: true,
  get: function () { return smtpEncryption.SmtpEncryption; }
});
Object.defineProperty(exports, "OrganizationKeyScopes", {
  enumerable: true,
  get: function () { return organizationKeyScopes.OrganizationKeyScopes; }
});
Object.defineProperty(exports, "Region", {
  enumerable: true,
  get: function () { return region.Region; }
});
Object.defineProperty(exports, "ProjectAuthMethodId", {
  enumerable: true,
  get: function () { return projectAuthMethodId.ProjectAuthMethodId; }
});
Object.defineProperty(exports, "ProjectOAuth2GooglePrompt", {
  enumerable: true,
  get: function () { return projectOAuth2GooglePrompt.ProjectOAuth2GooglePrompt; }
});
Object.defineProperty(exports, "ProjectOAuth2OidcPrompt", {
  enumerable: true,
  get: function () { return projectOAuth2OidcPrompt.ProjectOAuth2OidcPrompt; }
});
Object.defineProperty(exports, "ProjectOAuthProviderId", {
  enumerable: true,
  get: function () { return projectOAuthProviderId.ProjectOAuthProviderId; }
});
Object.defineProperty(exports, "ProjectPolicyId", {
  enumerable: true,
  get: function () { return projectPolicyId.ProjectPolicyId; }
});
Object.defineProperty(exports, "ProjectProtocolId", {
  enumerable: true,
  get: function () { return projectProtocolId.ProjectProtocolId; }
});
Object.defineProperty(exports, "ProjectServiceId", {
  enumerable: true,
  get: function () { return projectServiceId.ProjectServiceId; }
});
Object.defineProperty(exports, "ProjectSMTPSecure", {
  enumerable: true,
  get: function () { return projectSmtpSecure.ProjectSMTPSecure; }
});
Object.defineProperty(exports, "ProjectEmailTemplateId", {
  enumerable: true,
  get: function () { return projectEmailTemplateId.ProjectEmailTemplateId; }
});
Object.defineProperty(exports, "ProjectEmailTemplateLocale", {
  enumerable: true,
  get: function () { return projectEmailTemplateLocale.ProjectEmailTemplateLocale; }
});
Object.defineProperty(exports, "StatusCode", {
  enumerable: true,
  get: function () { return statusCode.StatusCode; }
});
Object.defineProperty(exports, "ProxyResourceType", {
  enumerable: true,
  get: function () { return proxyResourceType.ProxyResourceType; }
});
Object.defineProperty(exports, "Framework", {
  enumerable: true,
  get: function () { return framework.Framework; }
});
Object.defineProperty(exports, "BuildRuntime", {
  enumerable: true,
  get: function () { return buildRuntime.BuildRuntime; }
});
Object.defineProperty(exports, "Adapter", {
  enumerable: true,
  get: function () { return adapter.Adapter; }
});
Object.defineProperty(exports, "Compression", {
  enumerable: true,
  get: function () { return compression.Compression; }
});
Object.defineProperty(exports, "ImageGravity", {
  enumerable: true,
  get: function () { return imageGravity.ImageGravity; }
});
Object.defineProperty(exports, "TablesDBIndexType", {
  enumerable: true,
  get: function () { return tablesDbIndexType.TablesDBIndexType; }
});
Object.defineProperty(exports, "PasswordHash", {
  enumerable: true,
  get: function () { return passwordHash.PasswordHash; }
});
Object.defineProperty(exports, "MessagingProviderType", {
  enumerable: true,
  get: function () { return messagingProviderType.MessagingProviderType; }
});
Object.defineProperty(exports, "DatabaseType", {
  enumerable: true,
  get: function () { return databaseType.DatabaseType; }
});
Object.defineProperty(exports, "DatabaseStatus", {
  enumerable: true,
  get: function () { return databaseStatus.DatabaseStatus; }
});
Object.defineProperty(exports, "AttributeStatus", {
  enumerable: true,
  get: function () { return attributeStatus.AttributeStatus; }
});
Object.defineProperty(exports, "ColumnStatus", {
  enumerable: true,
  get: function () { return columnStatus.ColumnStatus; }
});
Object.defineProperty(exports, "IndexStatus", {
  enumerable: true,
  get: function () { return indexStatus.IndexStatus; }
});
Object.defineProperty(exports, "DeploymentStatus", {
  enumerable: true,
  get: function () { return deploymentStatus.DeploymentStatus; }
});
Object.defineProperty(exports, "ExecutionTrigger", {
  enumerable: true,
  get: function () { return executionTrigger.ExecutionTrigger; }
});
Object.defineProperty(exports, "ExecutionStatus", {
  enumerable: true,
  get: function () { return executionStatus.ExecutionStatus; }
});
Object.defineProperty(exports, "OAuth2GooglePrompt", {
  enumerable: true,
  get: function () { return oAuth2GooglePrompt.OAuth2GooglePrompt; }
});
Object.defineProperty(exports, "OAuth2OidcPrompt", {
  enumerable: true,
  get: function () { return oAuth2OidcPrompt.OAuth2OidcPrompt; }
});
Object.defineProperty(exports, "PlatformType", {
  enumerable: true,
  get: function () { return platformType.PlatformType; }
});
Object.defineProperty(exports, "ProxyRuleDeploymentResourceType", {
  enumerable: true,
  get: function () { return proxyRuleDeploymentResourceType.ProxyRuleDeploymentResourceType; }
});
Object.defineProperty(exports, "ProxyRuleStatus", {
  enumerable: true,
  get: function () { return proxyRuleStatus.ProxyRuleStatus; }
});
Object.defineProperty(exports, "MessageStatus", {
  enumerable: true,
  get: function () { return messageStatus.MessageStatus; }
});
Object.defineProperty(exports, "BillingPlanGroup", {
  enumerable: true,
  get: function () { return billingPlanGroup.BillingPlanGroup; }
});
