// node-appwrite/dist/inputFile.js
'use strict';

var undici = require('undici');

function isEdgeRuntime() {
  return typeof globalThis !== "undefined" && typeof globalThis.EdgeRuntime !== "undefined";
}
function assertFileSystemAvailable() {
  if (isEdgeRuntime()) {
    throw new Error("File system operations are not supported in edge runtimes. Please use InputFile.fromBuffer instead.");
  }
}
async function getFs() {
  assertFileSystemAvailable();
  try {
    const fs = await import('fs');
    return fs.promises;
  } catch {
    throw new Error("File system operations are not available in this runtime. Please use InputFile.fromBuffer instead.");
  }
}
function getFilename(path) {
  const segments = path.replace(/\\/g, "/").split("/").filter(Boolean);
  return segments.pop() ?? "file";
}
class InputFile {
  constructor(source, filename) {
    this.source = source;
    this.filename = filename;
  }
  static fromBuffer(parts, name) {
    if (parts && !ArrayBuffer.isView(parts) && typeof parts.arrayBuffer === "function") {
      return new InputFile({ type: "blob", data: parts }, name);
    }
    if (typeof parts === "string") {
      return new InputFile({ type: "buffer", data: new TextEncoder().encode(parts) }, name);
    }
    if (parts instanceof ArrayBuffer) {
      return new InputFile({ type: "buffer", data: new Uint8Array(parts) }, name);
    }
    if (ArrayBuffer.isView(parts)) {
      return new InputFile({
        type: "buffer",
        data: new Uint8Array(parts.buffer, parts.byteOffset, parts.byteLength)
      }, name);
    }
    throw new Error("Unsupported input type for InputFile.fromBuffer");
  }
  static fromPath(path, name) {
    assertFileSystemAvailable();
    return new InputFile({ type: "path", path }, name ?? getFilename(path));
  }
  static fromPlainText(content, name) {
    return new InputFile({ type: "buffer", data: new TextEncoder().encode(content) }, name);
  }
  async size() {
    switch (this.source.type) {
      case "path": {
        const fs = await getFs();
        return (await fs.stat(this.source.path)).size;
      }
      case "buffer":
        return this.source.data.length;
      case "blob":
        return this.source.data.size;
    }
  }
  async slice(start, end) {
    const length = end - start;
    switch (this.source.type) {
      case "path": {
        const fs = await getFs();
        const handle = await fs.open(this.source.path, "r");
        try {
          const buffer = new Uint8Array(length);
          const result = await handle.read(buffer, 0, length, start);
          return result.bytesRead === buffer.length ? buffer : buffer.subarray(0, result.bytesRead);
        } finally {
          await handle.close();
        }
      }
      case "buffer":
        return this.source.data.subarray(start, end);
      case "blob": {
        const arrayBuffer = await this.source.data.slice(start, end).arrayBuffer();
        return new Uint8Array(arrayBuffer);
      }
    }
  }
  async toFile() {
    const data = await this.toUint8Array();
    return new undici.File([data], this.filename);
  }
  async toUint8Array() {
    switch (this.source.type) {
      case "path": {
        const fs = await getFs();
        return await fs.readFile(this.source.path);
      }
      case "buffer":
        return this.source.data;
      case "blob":
        return new Uint8Array(await this.source.data.arrayBuffer());
    }
  }
}

exports.InputFile = InputFile;
