// node-appwrite/dist/enums/browser-permission.js
'use strict';

var BrowserPermission = /* @__PURE__ */ ((BrowserPermission2) => {
  BrowserPermission2["Geolocation"] = "geolocation";
  BrowserPermission2["Camera"] = "camera";
  BrowserPermission2["Microphone"] = "microphone";
  BrowserPermission2["Notifications"] = "notifications";
  BrowserPermission2["Midi"] = "midi";
  BrowserPermission2["Push"] = "push";
  BrowserPermission2["Clipboardread"] = "clipboard-read";
  BrowserPermission2["Clipboardwrite"] = "clipboard-write";
  BrowserPermission2["Paymenthandler"] = "payment-handler";
  BrowserPermission2["Usb"] = "usb";
  BrowserPermission2["Bluetooth"] = "bluetooth";
  BrowserPermission2["Accelerometer"] = "accelerometer";
  BrowserPermission2["Gyroscope"] = "gyroscope";
  BrowserPermission2["Magnetometer"] = "magnetometer";
  BrowserPermission2["Ambientlightsensor"] = "ambient-light-sensor";
  BrowserPermission2["Backgroundsync"] = "background-sync";
  BrowserPermission2["Persistentstorage"] = "persistent-storage";
  BrowserPermission2["Screenwakelock"] = "screen-wake-lock";
  BrowserPermission2["Webshare"] = "web-share";
  BrowserPermission2["Xrspatialtracking"] = "xr-spatial-tracking";
  return BrowserPermission2;
})(BrowserPermission || {});

exports.BrowserPermission = BrowserPermission;
