// node-appwrite/dist/enums/build-runtime.mjs
// src/enums/build-runtime.ts
var BuildRuntime = /* @__PURE__ */ ((BuildRuntime2) => {
  BuildRuntime2["Node145"] = "node-14.5";
  BuildRuntime2["Node160"] = "node-16.0";
  BuildRuntime2["Node180"] = "node-18.0";
  BuildRuntime2["Node190"] = "node-19.0";
  BuildRuntime2["Node200"] = "node-20.0";
  BuildRuntime2["Node210"] = "node-21.0";
  BuildRuntime2["Node22"] = "node-22";
  BuildRuntime2["Node23"] = "node-23";
  BuildRuntime2["Node24"] = "node-24";
  BuildRuntime2["Node25"] = "node-25";
  BuildRuntime2["Php80"] = "php-8.0";
  BuildRuntime2["Php81"] = "php-8.1";
  BuildRuntime2["Php82"] = "php-8.2";
  BuildRuntime2["Php83"] = "php-8.3";
  BuildRuntime2["Php84"] = "php-8.4";
  BuildRuntime2["Ruby30"] = "ruby-3.0";
  BuildRuntime2["Ruby31"] = "ruby-3.1";
  BuildRuntime2["Ruby32"] = "ruby-3.2";
  BuildRuntime2["Ruby33"] = "ruby-3.3";
  BuildRuntime2["Ruby34"] = "ruby-3.4";
  BuildRuntime2["Ruby40"] = "ruby-4.0";
  BuildRuntime2["Python38"] = "python-3.8";
  BuildRuntime2["Python39"] = "python-3.9";
  BuildRuntime2["Python310"] = "python-3.10";
  BuildRuntime2["Python311"] = "python-3.11";
  BuildRuntime2["Python312"] = "python-3.12";
  BuildRuntime2["Python313"] = "python-3.13";
  BuildRuntime2["Python314"] = "python-3.14";
  BuildRuntime2["Pythonml311"] = "python-ml-3.11";
  BuildRuntime2["Pythonml312"] = "python-ml-3.12";
  BuildRuntime2["Pythonml313"] = "python-ml-3.13";
  BuildRuntime2["Deno140"] = "deno-1.40";
  BuildRuntime2["Deno146"] = "deno-1.46";
  BuildRuntime2["Deno20"] = "deno-2.0";
  BuildRuntime2["Deno25"] = "deno-2.5";
  BuildRuntime2["Deno26"] = "deno-2.6";
  BuildRuntime2["Dart215"] = "dart-2.15";
  BuildRuntime2["Dart216"] = "dart-2.16";
  BuildRuntime2["Dart217"] = "dart-2.17";
  BuildRuntime2["Dart218"] = "dart-2.18";
  BuildRuntime2["Dart219"] = "dart-2.19";
  BuildRuntime2["Dart30"] = "dart-3.0";
  BuildRuntime2["Dart31"] = "dart-3.1";
  BuildRuntime2["Dart33"] = "dart-3.3";
  BuildRuntime2["Dart35"] = "dart-3.5";
  BuildRuntime2["Dart38"] = "dart-3.8";
  BuildRuntime2["Dart39"] = "dart-3.9";
  BuildRuntime2["Dart310"] = "dart-3.10";
  BuildRuntime2["Dart311"] = "dart-3.11";
  BuildRuntime2["Dart312"] = "dart-3.12";
  BuildRuntime2["Dotnet60"] = "dotnet-6.0";
  BuildRuntime2["Dotnet70"] = "dotnet-7.0";
  BuildRuntime2["Dotnet80"] = "dotnet-8.0";
  BuildRuntime2["Dotnet10"] = "dotnet-10";
  BuildRuntime2["Java80"] = "java-8.0";
  BuildRuntime2["Java110"] = "java-11.0";
  BuildRuntime2["Java170"] = "java-17.0";
  BuildRuntime2["Java180"] = "java-18.0";
  BuildRuntime2["Java210"] = "java-21.0";
  BuildRuntime2["Java22"] = "java-22";
  BuildRuntime2["Java25"] = "java-25";
  BuildRuntime2["Swift55"] = "swift-5.5";
  BuildRuntime2["Swift58"] = "swift-5.8";
  BuildRuntime2["Swift59"] = "swift-5.9";
  BuildRuntime2["Swift510"] = "swift-5.10";
  BuildRuntime2["Swift62"] = "swift-6.2";
  BuildRuntime2["Kotlin16"] = "kotlin-1.6";
  BuildRuntime2["Kotlin18"] = "kotlin-1.8";
  BuildRuntime2["Kotlin19"] = "kotlin-1.9";
  BuildRuntime2["Kotlin20"] = "kotlin-2.0";
  BuildRuntime2["Kotlin23"] = "kotlin-2.3";
  BuildRuntime2["Cpp17"] = "cpp-17";
  BuildRuntime2["Cpp20"] = "cpp-20";
  BuildRuntime2["Bun10"] = "bun-1.0";
  BuildRuntime2["Bun11"] = "bun-1.1";
  BuildRuntime2["Bun12"] = "bun-1.2";
  BuildRuntime2["Bun13"] = "bun-1.3";
  BuildRuntime2["Go123"] = "go-1.23";
  BuildRuntime2["Go124"] = "go-1.24";
  BuildRuntime2["Go125"] = "go-1.25";
  BuildRuntime2["Go126"] = "go-1.26";
  BuildRuntime2["Rust183"] = "rust-1.83";
  BuildRuntime2["Static1"] = "static-1";
  BuildRuntime2["Flutter324"] = "flutter-3.24";
  BuildRuntime2["Flutter327"] = "flutter-3.27";
  BuildRuntime2["Flutter329"] = "flutter-3.29";
  BuildRuntime2["Flutter332"] = "flutter-3.32";
  BuildRuntime2["Flutter335"] = "flutter-3.35";
  BuildRuntime2["Flutter338"] = "flutter-3.38";
  BuildRuntime2["Flutter341"] = "flutter-3.41";
  BuildRuntime2["Flutter344"] = "flutter-3.44";
  return BuildRuntime2;
})(BuildRuntime || {});

export { BuildRuntime };
