// node-appwrite/dist/enums/organization-key-scopes.mjs
// src/enums/organization-key-scopes.ts
var OrganizationKeyScopes = /* @__PURE__ */ ((OrganizationKeyScopes2) => {
  OrganizationKeyScopes2["ProjectsRead"] = "projects.read";
  OrganizationKeyScopes2["ProjectsWrite"] = "projects.write";
  OrganizationKeyScopes2["DevKeysRead"] = "devKeys.read";
  OrganizationKeyScopes2["DevKeysWrite"] = "devKeys.write";
  OrganizationKeyScopes2["OrganizationKeysRead"] = "organization.keys.read";
  OrganizationKeyScopes2["OrganizationKeysWrite"] = "organization.keys.write";
  OrganizationKeyScopes2["OrganizationMembershipsRead"] = "organization.memberships.read";
  OrganizationKeyScopes2["OrganizationMembershipsWrite"] = "organization.memberships.write";
  OrganizationKeyScopes2["OrganizationRead"] = "organization.read";
  OrganizationKeyScopes2["OrganizationWrite"] = "organization.write";
  OrganizationKeyScopes2["DomainsRead"] = "domains.read";
  OrganizationKeyScopes2["DomainsWrite"] = "domains.write";
  OrganizationKeyScopes2["KeysRead"] = "keys.read";
  OrganizationKeyScopes2["KeysWrite"] = "keys.write";
  return OrganizationKeyScopes2;
})(OrganizationKeyScopes || {});

export { OrganizationKeyScopes };
