// node-appwrite/dist/enums/project-key-scopes.js
'use strict';

var ProjectKeyScopes = /* @__PURE__ */ ((ProjectKeyScopes2) => {
  ProjectKeyScopes2["ProjectRead"] = "project.read";
  ProjectKeyScopes2["ProjectWrite"] = "project.write";
  ProjectKeyScopes2["KeysRead"] = "keys.read";
  ProjectKeyScopes2["KeysWrite"] = "keys.write";
  ProjectKeyScopes2["PlatformsRead"] = "platforms.read";
  ProjectKeyScopes2["PlatformsWrite"] = "platforms.write";
  ProjectKeyScopes2["MocksRead"] = "mocks.read";
  ProjectKeyScopes2["MocksWrite"] = "mocks.write";
  ProjectKeyScopes2["PoliciesRead"] = "policies.read";
  ProjectKeyScopes2["PoliciesWrite"] = "policies.write";
  ProjectKeyScopes2["ProjectPoliciesRead"] = "project.policies.read";
  ProjectKeyScopes2["ProjectPoliciesWrite"] = "project.policies.write";
  ProjectKeyScopes2["ProjectOauth2Read"] = "project.oauth2.read";
  ProjectKeyScopes2["ProjectOauth2Write"] = "project.oauth2.write";
  ProjectKeyScopes2["TemplatesRead"] = "templates.read";
  ProjectKeyScopes2["TemplatesWrite"] = "templates.write";
  ProjectKeyScopes2["StagesRead"] = "stages.read";
  ProjectKeyScopes2["StagesWrite"] = "stages.write";
  ProjectKeyScopes2["UsersRead"] = "users.read";
  ProjectKeyScopes2["UsersWrite"] = "users.write";
  ProjectKeyScopes2["SessionsRead"] = "sessions.read";
  ProjectKeyScopes2["SessionsWrite"] = "sessions.write";
  ProjectKeyScopes2["TeamsRead"] = "teams.read";
  ProjectKeyScopes2["TeamsWrite"] = "teams.write";
  ProjectKeyScopes2["DatabasesRead"] = "databases.read";
  ProjectKeyScopes2["DatabasesWrite"] = "databases.write";
  ProjectKeyScopes2["TablesRead"] = "tables.read";
  ProjectKeyScopes2["TablesWrite"] = "tables.write";
  ProjectKeyScopes2["ColumnsRead"] = "columns.read";
  ProjectKeyScopes2["ColumnsWrite"] = "columns.write";
  ProjectKeyScopes2["IndexesRead"] = "indexes.read";
  ProjectKeyScopes2["IndexesWrite"] = "indexes.write";
  ProjectKeyScopes2["RowsRead"] = "rows.read";
  ProjectKeyScopes2["RowsWrite"] = "rows.write";
  ProjectKeyScopes2["CollectionsRead"] = "collections.read";
  ProjectKeyScopes2["CollectionsWrite"] = "collections.write";
  ProjectKeyScopes2["AttributesRead"] = "attributes.read";
  ProjectKeyScopes2["AttributesWrite"] = "attributes.write";
  ProjectKeyScopes2["DocumentsRead"] = "documents.read";
  ProjectKeyScopes2["DocumentsWrite"] = "documents.write";
  ProjectKeyScopes2["BucketsRead"] = "buckets.read";
  ProjectKeyScopes2["BucketsWrite"] = "buckets.write";
  ProjectKeyScopes2["FilesRead"] = "files.read";
  ProjectKeyScopes2["FilesWrite"] = "files.write";
  ProjectKeyScopes2["TokensRead"] = "tokens.read";
  ProjectKeyScopes2["TokensWrite"] = "tokens.write";
  ProjectKeyScopes2["FunctionsRead"] = "functions.read";
  ProjectKeyScopes2["FunctionsWrite"] = "functions.write";
  ProjectKeyScopes2["ExecutionsRead"] = "executions.read";
  ProjectKeyScopes2["ExecutionsWrite"] = "executions.write";
  ProjectKeyScopes2["ExecutionRead"] = "execution.read";
  ProjectKeyScopes2["ExecutionWrite"] = "execution.write";
  ProjectKeyScopes2["SitesRead"] = "sites.read";
  ProjectKeyScopes2["SitesWrite"] = "sites.write";
  ProjectKeyScopes2["LogRead"] = "log.read";
  ProjectKeyScopes2["LogWrite"] = "log.write";
  ProjectKeyScopes2["ProvidersRead"] = "providers.read";
  ProjectKeyScopes2["ProvidersWrite"] = "providers.write";
  ProjectKeyScopes2["TopicsRead"] = "topics.read";
  ProjectKeyScopes2["TopicsWrite"] = "topics.write";
  ProjectKeyScopes2["SubscribersRead"] = "subscribers.read";
  ProjectKeyScopes2["SubscribersWrite"] = "subscribers.write";
  ProjectKeyScopes2["TargetsRead"] = "targets.read";
  ProjectKeyScopes2["TargetsWrite"] = "targets.write";
  ProjectKeyScopes2["MessagesRead"] = "messages.read";
  ProjectKeyScopes2["MessagesWrite"] = "messages.write";
  ProjectKeyScopes2["RulesRead"] = "rules.read";
  ProjectKeyScopes2["RulesWrite"] = "rules.write";
  ProjectKeyScopes2["WebhooksRead"] = "webhooks.read";
  ProjectKeyScopes2["WebhooksWrite"] = "webhooks.write";
  ProjectKeyScopes2["LocaleRead"] = "locale.read";
  ProjectKeyScopes2["AvatarsRead"] = "avatars.read";
  ProjectKeyScopes2["HealthRead"] = "health.read";
  ProjectKeyScopes2["AssistantRead"] = "assistant.read";
  ProjectKeyScopes2["MigrationsRead"] = "migrations.read";
  ProjectKeyScopes2["MigrationsWrite"] = "migrations.write";
  ProjectKeyScopes2["SchedulesRead"] = "schedules.read";
  ProjectKeyScopes2["SchedulesWrite"] = "schedules.write";
  ProjectKeyScopes2["VcsRead"] = "vcs.read";
  ProjectKeyScopes2["VcsWrite"] = "vcs.write";
  ProjectKeyScopes2["InsightsRead"] = "insights.read";
  ProjectKeyScopes2["InsightsWrite"] = "insights.write";
  ProjectKeyScopes2["ReportsRead"] = "reports.read";
  ProjectKeyScopes2["ReportsWrite"] = "reports.write";
  ProjectKeyScopes2["PresencesRead"] = "presences.read";
  ProjectKeyScopes2["PresencesWrite"] = "presences.write";
  ProjectKeyScopes2["BackupsPoliciesRead"] = "backups.policies.read";
  ProjectKeyScopes2["BackupsPoliciesWrite"] = "backups.policies.write";
  ProjectKeyScopes2["ArchivesRead"] = "archives.read";
  ProjectKeyScopes2["ArchivesWrite"] = "archives.write";
  ProjectKeyScopes2["RestorationsRead"] = "restorations.read";
  ProjectKeyScopes2["RestorationsWrite"] = "restorations.write";
  ProjectKeyScopes2["DedicatedDatabasesExecute"] = "dedicatedDatabases.execute";
  ProjectKeyScopes2["DomainsRead"] = "domains.read";
  ProjectKeyScopes2["DomainsWrite"] = "domains.write";
  ProjectKeyScopes2["EventsRead"] = "events.read";
  ProjectKeyScopes2["AppsRead"] = "apps.read";
  ProjectKeyScopes2["AppsWrite"] = "apps.write";
  ProjectKeyScopes2["Oauth2Read"] = "oauth2.read";
  ProjectKeyScopes2["Oauth2Write"] = "oauth2.write";
  ProjectKeyScopes2["UsageRead"] = "usage.read";
  return ProjectKeyScopes2;
})(ProjectKeyScopes || {});

exports.ProjectKeyScopes = ProjectKeyScopes;
