// node-appwrite/dist/enums/project-email-template-locale.js
'use strict';

var ProjectEmailTemplateLocale = /* @__PURE__ */ ((ProjectEmailTemplateLocale2) => {
  ProjectEmailTemplateLocale2["Af"] = "af";
  ProjectEmailTemplateLocale2["Arae"] = "ar-ae";
  ProjectEmailTemplateLocale2["Arbh"] = "ar-bh";
  ProjectEmailTemplateLocale2["Ardz"] = "ar-dz";
  ProjectEmailTemplateLocale2["Areg"] = "ar-eg";
  ProjectEmailTemplateLocale2["Ariq"] = "ar-iq";
  ProjectEmailTemplateLocale2["Arjo"] = "ar-jo";
  ProjectEmailTemplateLocale2["Arkw"] = "ar-kw";
  ProjectEmailTemplateLocale2["Arlb"] = "ar-lb";
  ProjectEmailTemplateLocale2["Arly"] = "ar-ly";
  ProjectEmailTemplateLocale2["Arma"] = "ar-ma";
  ProjectEmailTemplateLocale2["Arom"] = "ar-om";
  ProjectEmailTemplateLocale2["Arqa"] = "ar-qa";
  ProjectEmailTemplateLocale2["Arsa"] = "ar-sa";
  ProjectEmailTemplateLocale2["Arsy"] = "ar-sy";
  ProjectEmailTemplateLocale2["Artn"] = "ar-tn";
  ProjectEmailTemplateLocale2["Arye"] = "ar-ye";
  ProjectEmailTemplateLocale2["As"] = "as";
  ProjectEmailTemplateLocale2["Az"] = "az";
  ProjectEmailTemplateLocale2["Be"] = "be";
  ProjectEmailTemplateLocale2["Bg"] = "bg";
  ProjectEmailTemplateLocale2["Bh"] = "bh";
  ProjectEmailTemplateLocale2["Bn"] = "bn";
  ProjectEmailTemplateLocale2["Bs"] = "bs";
  ProjectEmailTemplateLocale2["Ca"] = "ca";
  ProjectEmailTemplateLocale2["Cs"] = "cs";
  ProjectEmailTemplateLocale2["Cy"] = "cy";
  ProjectEmailTemplateLocale2["Da"] = "da";
  ProjectEmailTemplateLocale2["De"] = "de";
  ProjectEmailTemplateLocale2["Deat"] = "de-at";
  ProjectEmailTemplateLocale2["Dech"] = "de-ch";
  ProjectEmailTemplateLocale2["Deli"] = "de-li";
  ProjectEmailTemplateLocale2["Delu"] = "de-lu";
  ProjectEmailTemplateLocale2["El"] = "el";
  ProjectEmailTemplateLocale2["En"] = "en";
  ProjectEmailTemplateLocale2["Enau"] = "en-au";
  ProjectEmailTemplateLocale2["Enbz"] = "en-bz";
  ProjectEmailTemplateLocale2["Enca"] = "en-ca";
  ProjectEmailTemplateLocale2["Engb"] = "en-gb";
  ProjectEmailTemplateLocale2["Enie"] = "en-ie";
  ProjectEmailTemplateLocale2["Enjm"] = "en-jm";
  ProjectEmailTemplateLocale2["Ennz"] = "en-nz";
  ProjectEmailTemplateLocale2["Entt"] = "en-tt";
  ProjectEmailTemplateLocale2["Enus"] = "en-us";
  ProjectEmailTemplateLocale2["Enza"] = "en-za";
  ProjectEmailTemplateLocale2["Eo"] = "eo";
  ProjectEmailTemplateLocale2["Es"] = "es";
  ProjectEmailTemplateLocale2["Esar"] = "es-ar";
  ProjectEmailTemplateLocale2["Esbo"] = "es-bo";
  ProjectEmailTemplateLocale2["Escl"] = "es-cl";
  ProjectEmailTemplateLocale2["Esco"] = "es-co";
  ProjectEmailTemplateLocale2["Escr"] = "es-cr";
  ProjectEmailTemplateLocale2["Esdo"] = "es-do";
  ProjectEmailTemplateLocale2["Esec"] = "es-ec";
  ProjectEmailTemplateLocale2["Esgt"] = "es-gt";
  ProjectEmailTemplateLocale2["Eshn"] = "es-hn";
  ProjectEmailTemplateLocale2["Esmx"] = "es-mx";
  ProjectEmailTemplateLocale2["Esni"] = "es-ni";
  ProjectEmailTemplateLocale2["Espa"] = "es-pa";
  ProjectEmailTemplateLocale2["Espe"] = "es-pe";
  ProjectEmailTemplateLocale2["Espr"] = "es-pr";
  ProjectEmailTemplateLocale2["Espy"] = "es-py";
  ProjectEmailTemplateLocale2["Essv"] = "es-sv";
  ProjectEmailTemplateLocale2["Esuy"] = "es-uy";
  ProjectEmailTemplateLocale2["Esve"] = "es-ve";
  ProjectEmailTemplateLocale2["Et"] = "et";
  ProjectEmailTemplateLocale2["Eu"] = "eu";
  ProjectEmailTemplateLocale2["Fa"] = "fa";
  ProjectEmailTemplateLocale2["Fi"] = "fi";
  ProjectEmailTemplateLocale2["Fo"] = "fo";
  ProjectEmailTemplateLocale2["Fr"] = "fr";
  ProjectEmailTemplateLocale2["Frbe"] = "fr-be";
  ProjectEmailTemplateLocale2["Frca"] = "fr-ca";
  ProjectEmailTemplateLocale2["Frch"] = "fr-ch";
  ProjectEmailTemplateLocale2["Frlu"] = "fr-lu";
  ProjectEmailTemplateLocale2["Ga"] = "ga";
  ProjectEmailTemplateLocale2["Gd"] = "gd";
  ProjectEmailTemplateLocale2["He"] = "he";
  ProjectEmailTemplateLocale2["Hi"] = "hi";
  ProjectEmailTemplateLocale2["Hr"] = "hr";
  ProjectEmailTemplateLocale2["Hu"] = "hu";
  ProjectEmailTemplateLocale2["Id"] = "id";
  ProjectEmailTemplateLocale2["Is"] = "is";
  ProjectEmailTemplateLocale2["It"] = "it";
  ProjectEmailTemplateLocale2["Itch"] = "it-ch";
  ProjectEmailTemplateLocale2["Ja"] = "ja";
  ProjectEmailTemplateLocale2["Ji"] = "ji";
  ProjectEmailTemplateLocale2["Ko"] = "ko";
  ProjectEmailTemplateLocale2["Ku"] = "ku";
  ProjectEmailTemplateLocale2["Lt"] = "lt";
  ProjectEmailTemplateLocale2["Lv"] = "lv";
  ProjectEmailTemplateLocale2["Mk"] = "mk";
  ProjectEmailTemplateLocale2["Ml"] = "ml";
  ProjectEmailTemplateLocale2["Ms"] = "ms";
  ProjectEmailTemplateLocale2["Mt"] = "mt";
  ProjectEmailTemplateLocale2["Nb"] = "nb";
  ProjectEmailTemplateLocale2["Ne"] = "ne";
  ProjectEmailTemplateLocale2["Nl"] = "nl";
  ProjectEmailTemplateLocale2["Nlbe"] = "nl-be";
  ProjectEmailTemplateLocale2["Nn"] = "nn";
  ProjectEmailTemplateLocale2["No"] = "no";
  ProjectEmailTemplateLocale2["Pa"] = "pa";
  ProjectEmailTemplateLocale2["Pl"] = "pl";
  ProjectEmailTemplateLocale2["Pt"] = "pt";
  ProjectEmailTemplateLocale2["Ptbr"] = "pt-br";
  ProjectEmailTemplateLocale2["Rm"] = "rm";
  ProjectEmailTemplateLocale2["Ro"] = "ro";
  ProjectEmailTemplateLocale2["Romd"] = "ro-md";
  ProjectEmailTemplateLocale2["Ru"] = "ru";
  ProjectEmailTemplateLocale2["Rumd"] = "ru-md";
  ProjectEmailTemplateLocale2["Sb"] = "sb";
  ProjectEmailTemplateLocale2["Sk"] = "sk";
  ProjectEmailTemplateLocale2["Sl"] = "sl";
  ProjectEmailTemplateLocale2["Sq"] = "sq";
  ProjectEmailTemplateLocale2["Sr"] = "sr";
  ProjectEmailTemplateLocale2["Sv"] = "sv";
  ProjectEmailTemplateLocale2["Svfi"] = "sv-fi";
  ProjectEmailTemplateLocale2["Th"] = "th";
  ProjectEmailTemplateLocale2["Tn"] = "tn";
  ProjectEmailTemplateLocale2["Tr"] = "tr";
  ProjectEmailTemplateLocale2["Ts"] = "ts";
  ProjectEmailTemplateLocale2["Ua"] = "ua";
  ProjectEmailTemplateLocale2["Ur"] = "ur";
  ProjectEmailTemplateLocale2["Ve"] = "ve";
  ProjectEmailTemplateLocale2["Vi"] = "vi";
  ProjectEmailTemplateLocale2["Xh"] = "xh";
  ProjectEmailTemplateLocale2["Zhcn"] = "zh-cn";
  ProjectEmailTemplateLocale2["Zhhk"] = "zh-hk";
  ProjectEmailTemplateLocale2["Zhsg"] = "zh-sg";
  ProjectEmailTemplateLocale2["Zhtw"] = "zh-tw";
  ProjectEmailTemplateLocale2["Zu"] = "zu";
  return ProjectEmailTemplateLocale2;
})(ProjectEmailTemplateLocale || {});

exports.ProjectEmailTemplateLocale = ProjectEmailTemplateLocale;
