// node-appwrite/dist/enums/project-policy-id.mjs
// src/enums/project-policy-id.ts
var ProjectPolicyId = /* @__PURE__ */ ((ProjectPolicyId2) => {
  ProjectPolicyId2["Passworddictionary"] = "password-dictionary";
  ProjectPolicyId2["Passwordhistory"] = "password-history";
  ProjectPolicyId2["Passwordstrength"] = "password-strength";
  ProjectPolicyId2["Passwordpersonaldata"] = "password-personal-data";
  ProjectPolicyId2["Sessionalert"] = "session-alert";
  ProjectPolicyId2["Sessionduration"] = "session-duration";
  ProjectPolicyId2["Sessioninvalidation"] = "session-invalidation";
  ProjectPolicyId2["Sessionlimit"] = "session-limit";
  ProjectPolicyId2["Userlimit"] = "user-limit";
  ProjectPolicyId2["Membershipprivacy"] = "membership-privacy";
  ProjectPolicyId2["Denyaliasedemail"] = "deny-aliased-email";
  ProjectPolicyId2["Denydisposableemail"] = "deny-disposable-email";
  ProjectPolicyId2["Denyfreeemail"] = "deny-free-email";
  ProjectPolicyId2["Denycorporateemail"] = "deny-corporate-email";
  return ProjectPolicyId2;
})(ProjectPolicyId || {});

export { ProjectPolicyId };
