// node-appwrite/dist/enums/project-o-auth-provider-id.js
'use strict';

var ProjectOAuthProviderId = /* @__PURE__ */ ((ProjectOAuthProviderId2) => {
  ProjectOAuthProviderId2["Amazon"] = "amazon";
  ProjectOAuthProviderId2["Apple"] = "apple";
  ProjectOAuthProviderId2["Appwrite"] = "appwrite";
  ProjectOAuthProviderId2["Auth0"] = "auth0";
  ProjectOAuthProviderId2["Authentik"] = "authentik";
  ProjectOAuthProviderId2["Autodesk"] = "autodesk";
  ProjectOAuthProviderId2["Bitbucket"] = "bitbucket";
  ProjectOAuthProviderId2["Bitly"] = "bitly";
  ProjectOAuthProviderId2["Box"] = "box";
  ProjectOAuthProviderId2["Dailymotion"] = "dailymotion";
  ProjectOAuthProviderId2["Discord"] = "discord";
  ProjectOAuthProviderId2["Disqus"] = "disqus";
  ProjectOAuthProviderId2["Dropbox"] = "dropbox";
  ProjectOAuthProviderId2["Etsy"] = "etsy";
  ProjectOAuthProviderId2["Facebook"] = "facebook";
  ProjectOAuthProviderId2["Figma"] = "figma";
  ProjectOAuthProviderId2["Fusionauth"] = "fusionauth";
  ProjectOAuthProviderId2["Github"] = "github";
  ProjectOAuthProviderId2["Gitlab"] = "gitlab";
  ProjectOAuthProviderId2["Google"] = "google";
  ProjectOAuthProviderId2["Keycloak"] = "keycloak";
  ProjectOAuthProviderId2["Kick"] = "kick";
  ProjectOAuthProviderId2["Linkedin"] = "linkedin";
  ProjectOAuthProviderId2["Microsoft"] = "microsoft";
  ProjectOAuthProviderId2["Notion"] = "notion";
  ProjectOAuthProviderId2["Oidc"] = "oidc";
  ProjectOAuthProviderId2["Okta"] = "okta";
  ProjectOAuthProviderId2["Paypal"] = "paypal";
  ProjectOAuthProviderId2["PaypalSandbox"] = "paypalSandbox";
  ProjectOAuthProviderId2["Podio"] = "podio";
  ProjectOAuthProviderId2["Salesforce"] = "salesforce";
  ProjectOAuthProviderId2["Slack"] = "slack";
  ProjectOAuthProviderId2["Spotify"] = "spotify";
  ProjectOAuthProviderId2["Stripe"] = "stripe";
  ProjectOAuthProviderId2["Tradeshift"] = "tradeshift";
  ProjectOAuthProviderId2["TradeshiftBox"] = "tradeshiftBox";
  ProjectOAuthProviderId2["Twitch"] = "twitch";
  ProjectOAuthProviderId2["Wordpress"] = "wordpress";
  ProjectOAuthProviderId2["X"] = "x";
  ProjectOAuthProviderId2["Yahoo"] = "yahoo";
  ProjectOAuthProviderId2["Yammer"] = "yammer";
  ProjectOAuthProviderId2["Yandex"] = "yandex";
  ProjectOAuthProviderId2["Zoho"] = "zoho";
  ProjectOAuthProviderId2["Zoom"] = "zoom";
  return ProjectOAuthProviderId2;
})(ProjectOAuthProviderId || {});

exports.ProjectOAuthProviderId = ProjectOAuthProviderId;
