// node-appwrite/dist/enums/project-service-id.mjs
// src/enums/project-service-id.ts
var ProjectServiceId = /* @__PURE__ */ ((ProjectServiceId2) => {
  ProjectServiceId2["Account"] = "account";
  ProjectServiceId2["Avatars"] = "avatars";
  ProjectServiceId2["Databases"] = "databases";
  ProjectServiceId2["Tablesdb"] = "tablesdb";
  ProjectServiceId2["Locale"] = "locale";
  ProjectServiceId2["Health"] = "health";
  ProjectServiceId2["Project"] = "project";
  ProjectServiceId2["Storage"] = "storage";
  ProjectServiceId2["Teams"] = "teams";
  ProjectServiceId2["Users"] = "users";
  ProjectServiceId2["Vcs"] = "vcs";
  ProjectServiceId2["Sites"] = "sites";
  ProjectServiceId2["Functions"] = "functions";
  ProjectServiceId2["Proxy"] = "proxy";
  ProjectServiceId2["Graphql"] = "graphql";
  ProjectServiceId2["Migrations"] = "migrations";
  ProjectServiceId2["Messaging"] = "messaging";
  ProjectServiceId2["Advisor"] = "advisor";
  ProjectServiceId2["Oauth2"] = "oauth2";
  return ProjectServiceId2;
})(ProjectServiceId || {});

export { ProjectServiceId };
