// node-appwrite/dist/enums/runtime.js
'use strict';

var Runtime = /* @__PURE__ */ ((Runtime2) => {
  Runtime2["Node145"] = "node-14.5";
  Runtime2["Node160"] = "node-16.0";
  Runtime2["Node180"] = "node-18.0";
  Runtime2["Node190"] = "node-19.0";
  Runtime2["Node200"] = "node-20.0";
  Runtime2["Node210"] = "node-21.0";
  Runtime2["Node22"] = "node-22";
  Runtime2["Node23"] = "node-23";
  Runtime2["Node24"] = "node-24";
  Runtime2["Node25"] = "node-25";
  Runtime2["Php80"] = "php-8.0";
  Runtime2["Php81"] = "php-8.1";
  Runtime2["Php82"] = "php-8.2";
  Runtime2["Php83"] = "php-8.3";
  Runtime2["Php84"] = "php-8.4";
  Runtime2["Ruby30"] = "ruby-3.0";
  Runtime2["Ruby31"] = "ruby-3.1";
  Runtime2["Ruby32"] = "ruby-3.2";
  Runtime2["Ruby33"] = "ruby-3.3";
  Runtime2["Ruby34"] = "ruby-3.4";
  Runtime2["Ruby40"] = "ruby-4.0";
  Runtime2["Python38"] = "python-3.8";
  Runtime2["Python39"] = "python-3.9";
  Runtime2["Python310"] = "python-3.10";
  Runtime2["Python311"] = "python-3.11";
  Runtime2["Python312"] = "python-3.12";
  Runtime2["Python313"] = "python-3.13";
  Runtime2["Python314"] = "python-3.14";
  Runtime2["Pythonml311"] = "python-ml-3.11";
  Runtime2["Pythonml312"] = "python-ml-3.12";
  Runtime2["Pythonml313"] = "python-ml-3.13";
  Runtime2["Deno140"] = "deno-1.40";
  Runtime2["Deno146"] = "deno-1.46";
  Runtime2["Deno20"] = "deno-2.0";
  Runtime2["Deno25"] = "deno-2.5";
  Runtime2["Deno26"] = "deno-2.6";
  Runtime2["Dart215"] = "dart-2.15";
  Runtime2["Dart216"] = "dart-2.16";
  Runtime2["Dart217"] = "dart-2.17";
  Runtime2["Dart218"] = "dart-2.18";
  Runtime2["Dart219"] = "dart-2.19";
  Runtime2["Dart30"] = "dart-3.0";
  Runtime2["Dart31"] = "dart-3.1";
  Runtime2["Dart33"] = "dart-3.3";
  Runtime2["Dart35"] = "dart-3.5";
  Runtime2["Dart38"] = "dart-3.8";
  Runtime2["Dart39"] = "dart-3.9";
  Runtime2["Dart310"] = "dart-3.10";
  Runtime2["Dart311"] = "dart-3.11";
  Runtime2["Dart312"] = "dart-3.12";
  Runtime2["Dotnet60"] = "dotnet-6.0";
  Runtime2["Dotnet70"] = "dotnet-7.0";
  Runtime2["Dotnet80"] = "dotnet-8.0";
  Runtime2["Dotnet10"] = "dotnet-10";
  Runtime2["Java80"] = "java-8.0";
  Runtime2["Java110"] = "java-11.0";
  Runtime2["Java170"] = "java-17.0";
  Runtime2["Java180"] = "java-18.0";
  Runtime2["Java210"] = "java-21.0";
  Runtime2["Java22"] = "java-22";
  Runtime2["Java25"] = "java-25";
  Runtime2["Swift55"] = "swift-5.5";
  Runtime2["Swift58"] = "swift-5.8";
  Runtime2["Swift59"] = "swift-5.9";
  Runtime2["Swift510"] = "swift-5.10";
  Runtime2["Swift62"] = "swift-6.2";
  Runtime2["Kotlin16"] = "kotlin-1.6";
  Runtime2["Kotlin18"] = "kotlin-1.8";
  Runtime2["Kotlin19"] = "kotlin-1.9";
  Runtime2["Kotlin20"] = "kotlin-2.0";
  Runtime2["Kotlin23"] = "kotlin-2.3";
  Runtime2["Cpp17"] = "cpp-17";
  Runtime2["Cpp20"] = "cpp-20";
  Runtime2["Bun10"] = "bun-1.0";
  Runtime2["Bun11"] = "bun-1.1";
  Runtime2["Bun12"] = "bun-1.2";
  Runtime2["Bun13"] = "bun-1.3";
  Runtime2["Go123"] = "go-1.23";
  Runtime2["Go124"] = "go-1.24";
  Runtime2["Go125"] = "go-1.25";
  Runtime2["Go126"] = "go-1.26";
  Runtime2["Rust183"] = "rust-1.83";
  Runtime2["Static1"] = "static-1";
  Runtime2["Flutter324"] = "flutter-3.24";
  Runtime2["Flutter327"] = "flutter-3.27";
  Runtime2["Flutter329"] = "flutter-3.29";
  Runtime2["Flutter332"] = "flutter-3.32";
  Runtime2["Flutter335"] = "flutter-3.35";
  Runtime2["Flutter338"] = "flutter-3.38";
  Runtime2["Flutter341"] = "flutter-3.41";
  Runtime2["Flutter344"] = "flutter-3.44";
  return Runtime2;
})(Runtime || {});

exports.Runtime = Runtime;
