// node-appwrite/dist/enums/timezone.js
'use strict';

var Timezone = /* @__PURE__ */ ((Timezone2) => {
  Timezone2["AfricaAbidjan"] = "africa/abidjan";
  Timezone2["AfricaAccra"] = "africa/accra";
  Timezone2["AfricaAddisAbaba"] = "africa/addis_ababa";
  Timezone2["AfricaAlgiers"] = "africa/algiers";
  Timezone2["AfricaAsmara"] = "africa/asmara";
  Timezone2["AfricaBamako"] = "africa/bamako";
  Timezone2["AfricaBangui"] = "africa/bangui";
  Timezone2["AfricaBanjul"] = "africa/banjul";
  Timezone2["AfricaBissau"] = "africa/bissau";
  Timezone2["AfricaBlantyre"] = "africa/blantyre";
  Timezone2["AfricaBrazzaville"] = "africa/brazzaville";
  Timezone2["AfricaBujumbura"] = "africa/bujumbura";
  Timezone2["AfricaCairo"] = "africa/cairo";
  Timezone2["AfricaCasablanca"] = "africa/casablanca";
  Timezone2["AfricaCeuta"] = "africa/ceuta";
  Timezone2["AfricaConakry"] = "africa/conakry";
  Timezone2["AfricaDakar"] = "africa/dakar";
  Timezone2["AfricaDarEsSalaam"] = "africa/dar_es_salaam";
  Timezone2["AfricaDjibouti"] = "africa/djibouti";
  Timezone2["AfricaDouala"] = "africa/douala";
  Timezone2["AfricaElAaiun"] = "africa/el_aaiun";
  Timezone2["AfricaFreetown"] = "africa/freetown";
  Timezone2["AfricaGaborone"] = "africa/gaborone";
  Timezone2["AfricaHarare"] = "africa/harare";
  Timezone2["AfricaJohannesburg"] = "africa/johannesburg";
  Timezone2["AfricaJuba"] = "africa/juba";
  Timezone2["AfricaKampala"] = "africa/kampala";
  Timezone2["AfricaKhartoum"] = "africa/khartoum";
  Timezone2["AfricaKigali"] = "africa/kigali";
  Timezone2["AfricaKinshasa"] = "africa/kinshasa";
  Timezone2["AfricaLagos"] = "africa/lagos";
  Timezone2["AfricaLibreville"] = "africa/libreville";
  Timezone2["AfricaLome"] = "africa/lome";
  Timezone2["AfricaLuanda"] = "africa/luanda";
  Timezone2["AfricaLubumbashi"] = "africa/lubumbashi";
  Timezone2["AfricaLusaka"] = "africa/lusaka";
  Timezone2["AfricaMalabo"] = "africa/malabo";
  Timezone2["AfricaMaputo"] = "africa/maputo";
  Timezone2["AfricaMaseru"] = "africa/maseru";
  Timezone2["AfricaMbabane"] = "africa/mbabane";
  Timezone2["AfricaMogadishu"] = "africa/mogadishu";
  Timezone2["AfricaMonrovia"] = "africa/monrovia";
  Timezone2["AfricaNairobi"] = "africa/nairobi";
  Timezone2["AfricaNdjamena"] = "africa/ndjamena";
  Timezone2["AfricaNiamey"] = "africa/niamey";
  Timezone2["AfricaNouakchott"] = "africa/nouakchott";
  Timezone2["AfricaOuagadougou"] = "africa/ouagadougou";
  Timezone2["AfricaPortonovo"] = "africa/porto-novo";
  Timezone2["AfricaSaoTome"] = "africa/sao_tome";
  Timezone2["AfricaTripoli"] = "africa/tripoli";
  Timezone2["AfricaTunis"] = "africa/tunis";
  Timezone2["AfricaWindhoek"] = "africa/windhoek";
  Timezone2["AmericaAdak"] = "america/adak";
  Timezone2["AmericaAnchorage"] = "america/anchorage";
  Timezone2["AmericaAnguilla"] = "america/anguilla";
  Timezone2["AmericaAntigua"] = "america/antigua";
  Timezone2["AmericaAraguaina"] = "america/araguaina";
  Timezone2["AmericaArgentinaBuenosAires"] = "america/argentina/buenos_aires";
  Timezone2["AmericaArgentinaCatamarca"] = "america/argentina/catamarca";
  Timezone2["AmericaArgentinaCordoba"] = "america/argentina/cordoba";
  Timezone2["AmericaArgentinaJujuy"] = "america/argentina/jujuy";
  Timezone2["AmericaArgentinaLaRioja"] = "america/argentina/la_rioja";
  Timezone2["AmericaArgentinaMendoza"] = "america/argentina/mendoza";
  Timezone2["AmericaArgentinaRioGallegos"] = "america/argentina/rio_gallegos";
  Timezone2["AmericaArgentinaSalta"] = "america/argentina/salta";
  Timezone2["AmericaArgentinaSanJuan"] = "america/argentina/san_juan";
  Timezone2["AmericaArgentinaSanLuis"] = "america/argentina/san_luis";
  Timezone2["AmericaArgentinaTucuman"] = "america/argentina/tucuman";
  Timezone2["AmericaArgentinaUshuaia"] = "america/argentina/ushuaia";
  Timezone2["AmericaAruba"] = "america/aruba";
  Timezone2["AmericaAsuncion"] = "america/asuncion";
  Timezone2["AmericaAtikokan"] = "america/atikokan";
  Timezone2["AmericaBahia"] = "america/bahia";
  Timezone2["AmericaBahiaBanderas"] = "america/bahia_banderas";
  Timezone2["AmericaBarbados"] = "america/barbados";
  Timezone2["AmericaBelem"] = "america/belem";
  Timezone2["AmericaBelize"] = "america/belize";
  Timezone2["AmericaBlancsablon"] = "america/blanc-sablon";
  Timezone2["AmericaBoaVista"] = "america/boa_vista";
  Timezone2["AmericaBogota"] = "america/bogota";
  Timezone2["AmericaBoise"] = "america/boise";
  Timezone2["AmericaCambridgeBay"] = "america/cambridge_bay";
  Timezone2["AmericaCampoGrande"] = "america/campo_grande";
  Timezone2["AmericaCancun"] = "america/cancun";
  Timezone2["AmericaCaracas"] = "america/caracas";
  Timezone2["AmericaCayenne"] = "america/cayenne";
  Timezone2["AmericaCayman"] = "america/cayman";
  Timezone2["AmericaChicago"] = "america/chicago";
  Timezone2["AmericaChihuahua"] = "america/chihuahua";
  Timezone2["AmericaCiudadJuarez"] = "america/ciudad_juarez";
  Timezone2["AmericaCostaRica"] = "america/costa_rica";
  Timezone2["AmericaCoyhaique"] = "america/coyhaique";
  Timezone2["AmericaCreston"] = "america/creston";
  Timezone2["AmericaCuiaba"] = "america/cuiaba";
  Timezone2["AmericaCuracao"] = "america/curacao";
  Timezone2["AmericaDanmarkshavn"] = "america/danmarkshavn";
  Timezone2["AmericaDawson"] = "america/dawson";
  Timezone2["AmericaDawsonCreek"] = "america/dawson_creek";
  Timezone2["AmericaDenver"] = "america/denver";
  Timezone2["AmericaDetroit"] = "america/detroit";
  Timezone2["AmericaDominica"] = "america/dominica";
  Timezone2["AmericaEdmonton"] = "america/edmonton";
  Timezone2["AmericaEirunepe"] = "america/eirunepe";
  Timezone2["AmericaElSalvador"] = "america/el_salvador";
  Timezone2["AmericaFortNelson"] = "america/fort_nelson";
  Timezone2["AmericaFortaleza"] = "america/fortaleza";
  Timezone2["AmericaGlaceBay"] = "america/glace_bay";
  Timezone2["AmericaGooseBay"] = "america/goose_bay";
  Timezone2["AmericaGrandTurk"] = "america/grand_turk";
  Timezone2["AmericaGrenada"] = "america/grenada";
  Timezone2["AmericaGuadeloupe"] = "america/guadeloupe";
  Timezone2["AmericaGuatemala"] = "america/guatemala";
  Timezone2["AmericaGuayaquil"] = "america/guayaquil";
  Timezone2["AmericaGuyana"] = "america/guyana";
  Timezone2["AmericaHalifax"] = "america/halifax";
  Timezone2["AmericaHavana"] = "america/havana";
  Timezone2["AmericaHermosillo"] = "america/hermosillo";
  Timezone2["AmericaIndianaIndianapolis"] = "america/indiana/indianapolis";
  Timezone2["AmericaIndianaKnox"] = "america/indiana/knox";
  Timezone2["AmericaIndianaMarengo"] = "america/indiana/marengo";
  Timezone2["AmericaIndianaPetersburg"] = "america/indiana/petersburg";
  Timezone2["AmericaIndianaTellCity"] = "america/indiana/tell_city";
  Timezone2["AmericaIndianaVevay"] = "america/indiana/vevay";
  Timezone2["AmericaIndianaVincennes"] = "america/indiana/vincennes";
  Timezone2["AmericaIndianaWinamac"] = "america/indiana/winamac";
  Timezone2["AmericaInuvik"] = "america/inuvik";
  Timezone2["AmericaIqaluit"] = "america/iqaluit";
  Timezone2["AmericaJamaica"] = "america/jamaica";
  Timezone2["AmericaJuneau"] = "america/juneau";
  Timezone2["AmericaKentuckyLouisville"] = "america/kentucky/louisville";
  Timezone2["AmericaKentuckyMonticello"] = "america/kentucky/monticello";
  Timezone2["AmericaKralendijk"] = "america/kralendijk";
  Timezone2["AmericaLaPaz"] = "america/la_paz";
  Timezone2["AmericaLima"] = "america/lima";
  Timezone2["AmericaLosAngeles"] = "america/los_angeles";
  Timezone2["AmericaLowerPrinces"] = "america/lower_princes";
  Timezone2["AmericaMaceio"] = "america/maceio";
  Timezone2["AmericaManagua"] = "america/managua";
  Timezone2["AmericaManaus"] = "america/manaus";
  Timezone2["AmericaMarigot"] = "america/marigot";
  Timezone2["AmericaMartinique"] = "america/martinique";
  Timezone2["AmericaMatamoros"] = "america/matamoros";
  Timezone2["AmericaMazatlan"] = "america/mazatlan";
  Timezone2["AmericaMenominee"] = "america/menominee";
  Timezone2["AmericaMerida"] = "america/merida";
  Timezone2["AmericaMetlakatla"] = "america/metlakatla";
  Timezone2["AmericaMexicoCity"] = "america/mexico_city";
  Timezone2["AmericaMiquelon"] = "america/miquelon";
  Timezone2["AmericaMoncton"] = "america/moncton";
  Timezone2["AmericaMonterrey"] = "america/monterrey";
  Timezone2["AmericaMontevideo"] = "america/montevideo";
  Timezone2["AmericaMontserrat"] = "america/montserrat";
  Timezone2["AmericaNassau"] = "america/nassau";
  Timezone2["AmericaNewYork"] = "america/new_york";
  Timezone2["AmericaNome"] = "america/nome";
  Timezone2["AmericaNoronha"] = "america/noronha";
  Timezone2["AmericaNorthDakotaBeulah"] = "america/north_dakota/beulah";
  Timezone2["AmericaNorthDakotaCenter"] = "america/north_dakota/center";
  Timezone2["AmericaNorthDakotaNewSalem"] = "america/north_dakota/new_salem";
  Timezone2["AmericaNuuk"] = "america/nuuk";
  Timezone2["AmericaOjinaga"] = "america/ojinaga";
  Timezone2["AmericaPanama"] = "america/panama";
  Timezone2["AmericaParamaribo"] = "america/paramaribo";
  Timezone2["AmericaPhoenix"] = "america/phoenix";
  Timezone2["AmericaPortauprince"] = "america/port-au-prince";
  Timezone2["AmericaPortOfSpain"] = "america/port_of_spain";
  Timezone2["AmericaPortoVelho"] = "america/porto_velho";
  Timezone2["AmericaPuertoRico"] = "america/puerto_rico";
  Timezone2["AmericaPuntaArenas"] = "america/punta_arenas";
  Timezone2["AmericaRankinInlet"] = "america/rankin_inlet";
  Timezone2["AmericaRecife"] = "america/recife";
  Timezone2["AmericaRegina"] = "america/regina";
  Timezone2["AmericaResolute"] = "america/resolute";
  Timezone2["AmericaRioBranco"] = "america/rio_branco";
  Timezone2["AmericaSantarem"] = "america/santarem";
  Timezone2["AmericaSantiago"] = "america/santiago";
  Timezone2["AmericaSantoDomingo"] = "america/santo_domingo";
  Timezone2["AmericaSaoPaulo"] = "america/sao_paulo";
  Timezone2["AmericaScoresbysund"] = "america/scoresbysund";
  Timezone2["AmericaSitka"] = "america/sitka";
  Timezone2["AmericaStBarthelemy"] = "america/st_barthelemy";
  Timezone2["AmericaStJohns"] = "america/st_johns";
  Timezone2["AmericaStKitts"] = "america/st_kitts";
  Timezone2["AmericaStLucia"] = "america/st_lucia";
  Timezone2["AmericaStThomas"] = "america/st_thomas";
  Timezone2["AmericaStVincent"] = "america/st_vincent";
  Timezone2["AmericaSwiftCurrent"] = "america/swift_current";
  Timezone2["AmericaTegucigalpa"] = "america/tegucigalpa";
  Timezone2["AmericaThule"] = "america/thule";
  Timezone2["AmericaTijuana"] = "america/tijuana";
  Timezone2["AmericaToronto"] = "america/toronto";
  Timezone2["AmericaTortola"] = "america/tortola";
  Timezone2["AmericaVancouver"] = "america/vancouver";
  Timezone2["AmericaWhitehorse"] = "america/whitehorse";
  Timezone2["AmericaWinnipeg"] = "america/winnipeg";
  Timezone2["AmericaYakutat"] = "america/yakutat";
  Timezone2["AntarcticaCasey"] = "antarctica/casey";
  Timezone2["AntarcticaDavis"] = "antarctica/davis";
  Timezone2["AntarcticaDumontdurville"] = "antarctica/dumontdurville";
  Timezone2["AntarcticaMacquarie"] = "antarctica/macquarie";
  Timezone2["AntarcticaMawson"] = "antarctica/mawson";
  Timezone2["AntarcticaMcmurdo"] = "antarctica/mcmurdo";
  Timezone2["AntarcticaPalmer"] = "antarctica/palmer";
  Timezone2["AntarcticaRothera"] = "antarctica/rothera";
  Timezone2["AntarcticaSyowa"] = "antarctica/syowa";
  Timezone2["AntarcticaTroll"] = "antarctica/troll";
  Timezone2["AntarcticaVostok"] = "antarctica/vostok";
  Timezone2["ArcticLongyearbyen"] = "arctic/longyearbyen";
  Timezone2["AsiaAden"] = "asia/aden";
  Timezone2["AsiaAlmaty"] = "asia/almaty";
  Timezone2["AsiaAmman"] = "asia/amman";
  Timezone2["AsiaAnadyr"] = "asia/anadyr";
  Timezone2["AsiaAqtau"] = "asia/aqtau";
  Timezone2["AsiaAqtobe"] = "asia/aqtobe";
  Timezone2["AsiaAshgabat"] = "asia/ashgabat";
  Timezone2["AsiaAtyrau"] = "asia/atyrau";
  Timezone2["AsiaBaghdad"] = "asia/baghdad";
  Timezone2["AsiaBahrain"] = "asia/bahrain";
  Timezone2["AsiaBaku"] = "asia/baku";
  Timezone2["AsiaBangkok"] = "asia/bangkok";
  Timezone2["AsiaBarnaul"] = "asia/barnaul";
  Timezone2["AsiaBeirut"] = "asia/beirut";
  Timezone2["AsiaBishkek"] = "asia/bishkek";
  Timezone2["AsiaBrunei"] = "asia/brunei";
  Timezone2["AsiaChita"] = "asia/chita";
  Timezone2["AsiaColombo"] = "asia/colombo";
  Timezone2["AsiaDamascus"] = "asia/damascus";
  Timezone2["AsiaDhaka"] = "asia/dhaka";
  Timezone2["AsiaDili"] = "asia/dili";
  Timezone2["AsiaDubai"] = "asia/dubai";
  Timezone2["AsiaDushanbe"] = "asia/dushanbe";
  Timezone2["AsiaFamagusta"] = "asia/famagusta";
  Timezone2["AsiaGaza"] = "asia/gaza";
  Timezone2["AsiaHebron"] = "asia/hebron";
  Timezone2["AsiaHoChiMinh"] = "asia/ho_chi_minh";
  Timezone2["AsiaHongKong"] = "asia/hong_kong";
  Timezone2["AsiaHovd"] = "asia/hovd";
  Timezone2["AsiaIrkutsk"] = "asia/irkutsk";
  Timezone2["AsiaJakarta"] = "asia/jakarta";
  Timezone2["AsiaJayapura"] = "asia/jayapura";
  Timezone2["AsiaJerusalem"] = "asia/jerusalem";
  Timezone2["AsiaKabul"] = "asia/kabul";
  Timezone2["AsiaKamchatka"] = "asia/kamchatka";
  Timezone2["AsiaKarachi"] = "asia/karachi";
  Timezone2["AsiaKathmandu"] = "asia/kathmandu";
  Timezone2["AsiaKhandyga"] = "asia/khandyga";
  Timezone2["AsiaKolkata"] = "asia/kolkata";
  Timezone2["AsiaKrasnoyarsk"] = "asia/krasnoyarsk";
  Timezone2["AsiaKualaLumpur"] = "asia/kuala_lumpur";
  Timezone2["AsiaKuching"] = "asia/kuching";
  Timezone2["AsiaKuwait"] = "asia/kuwait";
  Timezone2["AsiaMacau"] = "asia/macau";
  Timezone2["AsiaMagadan"] = "asia/magadan";
  Timezone2["AsiaMakassar"] = "asia/makassar";
  Timezone2["AsiaManila"] = "asia/manila";
  Timezone2["AsiaMuscat"] = "asia/muscat";
  Timezone2["AsiaNicosia"] = "asia/nicosia";
  Timezone2["AsiaNovokuznetsk"] = "asia/novokuznetsk";
  Timezone2["AsiaNovosibirsk"] = "asia/novosibirsk";
  Timezone2["AsiaOmsk"] = "asia/omsk";
  Timezone2["AsiaOral"] = "asia/oral";
  Timezone2["AsiaPhnomPenh"] = "asia/phnom_penh";
  Timezone2["AsiaPontianak"] = "asia/pontianak";
  Timezone2["AsiaPyongyang"] = "asia/pyongyang";
  Timezone2["AsiaQatar"] = "asia/qatar";
  Timezone2["AsiaQostanay"] = "asia/qostanay";
  Timezone2["AsiaQyzylorda"] = "asia/qyzylorda";
  Timezone2["AsiaRiyadh"] = "asia/riyadh";
  Timezone2["AsiaSakhalin"] = "asia/sakhalin";
  Timezone2["AsiaSamarkand"] = "asia/samarkand";
  Timezone2["AsiaSeoul"] = "asia/seoul";
  Timezone2["AsiaShanghai"] = "asia/shanghai";
  Timezone2["AsiaSingapore"] = "asia/singapore";
  Timezone2["AsiaSrednekolymsk"] = "asia/srednekolymsk";
  Timezone2["AsiaTaipei"] = "asia/taipei";
  Timezone2["AsiaTashkent"] = "asia/tashkent";
  Timezone2["AsiaTbilisi"] = "asia/tbilisi";
  Timezone2["AsiaTehran"] = "asia/tehran";
  Timezone2["AsiaThimphu"] = "asia/thimphu";
  Timezone2["AsiaTokyo"] = "asia/tokyo";
  Timezone2["AsiaTomsk"] = "asia/tomsk";
  Timezone2["AsiaUlaanbaatar"] = "asia/ulaanbaatar";
  Timezone2["AsiaUrumqi"] = "asia/urumqi";
  Timezone2["AsiaUstnera"] = "asia/ust-nera";
  Timezone2["AsiaVientiane"] = "asia/vientiane";
  Timezone2["AsiaVladivostok"] = "asia/vladivostok";
  Timezone2["AsiaYakutsk"] = "asia/yakutsk";
  Timezone2["AsiaYangon"] = "asia/yangon";
  Timezone2["AsiaYekaterinburg"] = "asia/yekaterinburg";
  Timezone2["AsiaYerevan"] = "asia/yerevan";
  Timezone2["AtlanticAzores"] = "atlantic/azores";
  Timezone2["AtlanticBermuda"] = "atlantic/bermuda";
  Timezone2["AtlanticCanary"] = "atlantic/canary";
  Timezone2["AtlanticCapeVerde"] = "atlantic/cape_verde";
  Timezone2["AtlanticFaroe"] = "atlantic/faroe";
  Timezone2["AtlanticMadeira"] = "atlantic/madeira";
  Timezone2["AtlanticReykjavik"] = "atlantic/reykjavik";
  Timezone2["AtlanticSouthGeorgia"] = "atlantic/south_georgia";
  Timezone2["AtlanticStHelena"] = "atlantic/st_helena";
  Timezone2["AtlanticStanley"] = "atlantic/stanley";
  Timezone2["AustraliaAdelaide"] = "australia/adelaide";
  Timezone2["AustraliaBrisbane"] = "australia/brisbane";
  Timezone2["AustraliaBrokenHill"] = "australia/broken_hill";
  Timezone2["AustraliaDarwin"] = "australia/darwin";
  Timezone2["AustraliaEucla"] = "australia/eucla";
  Timezone2["AustraliaHobart"] = "australia/hobart";
  Timezone2["AustraliaLindeman"] = "australia/lindeman";
  Timezone2["AustraliaLordHowe"] = "australia/lord_howe";
  Timezone2["AustraliaMelbourne"] = "australia/melbourne";
  Timezone2["AustraliaPerth"] = "australia/perth";
  Timezone2["AustraliaSydney"] = "australia/sydney";
  Timezone2["EuropeAmsterdam"] = "europe/amsterdam";
  Timezone2["EuropeAndorra"] = "europe/andorra";
  Timezone2["EuropeAstrakhan"] = "europe/astrakhan";
  Timezone2["EuropeAthens"] = "europe/athens";
  Timezone2["EuropeBelgrade"] = "europe/belgrade";
  Timezone2["EuropeBerlin"] = "europe/berlin";
  Timezone2["EuropeBratislava"] = "europe/bratislava";
  Timezone2["EuropeBrussels"] = "europe/brussels";
  Timezone2["EuropeBucharest"] = "europe/bucharest";
  Timezone2["EuropeBudapest"] = "europe/budapest";
  Timezone2["EuropeBusingen"] = "europe/busingen";
  Timezone2["EuropeChisinau"] = "europe/chisinau";
  Timezone2["EuropeCopenhagen"] = "europe/copenhagen";
  Timezone2["EuropeDublin"] = "europe/dublin";
  Timezone2["EuropeGibraltar"] = "europe/gibraltar";
  Timezone2["EuropeGuernsey"] = "europe/guernsey";
  Timezone2["EuropeHelsinki"] = "europe/helsinki";
  Timezone2["EuropeIsleOfMan"] = "europe/isle_of_man";
  Timezone2["EuropeIstanbul"] = "europe/istanbul";
  Timezone2["EuropeJersey"] = "europe/jersey";
  Timezone2["EuropeKaliningrad"] = "europe/kaliningrad";
  Timezone2["EuropeKirov"] = "europe/kirov";
  Timezone2["EuropeKyiv"] = "europe/kyiv";
  Timezone2["EuropeLisbon"] = "europe/lisbon";
  Timezone2["EuropeLjubljana"] = "europe/ljubljana";
  Timezone2["EuropeLondon"] = "europe/london";
  Timezone2["EuropeLuxembourg"] = "europe/luxembourg";
  Timezone2["EuropeMadrid"] = "europe/madrid";
  Timezone2["EuropeMalta"] = "europe/malta";
  Timezone2["EuropeMariehamn"] = "europe/mariehamn";
  Timezone2["EuropeMinsk"] = "europe/minsk";
  Timezone2["EuropeMonaco"] = "europe/monaco";
  Timezone2["EuropeMoscow"] = "europe/moscow";
  Timezone2["EuropeOslo"] = "europe/oslo";
  Timezone2["EuropeParis"] = "europe/paris";
  Timezone2["EuropePodgorica"] = "europe/podgorica";
  Timezone2["EuropePrague"] = "europe/prague";
  Timezone2["EuropeRiga"] = "europe/riga";
  Timezone2["EuropeRome"] = "europe/rome";
  Timezone2["EuropeSamara"] = "europe/samara";
  Timezone2["EuropeSanMarino"] = "europe/san_marino";
  Timezone2["EuropeSarajevo"] = "europe/sarajevo";
  Timezone2["EuropeSaratov"] = "europe/saratov";
  Timezone2["EuropeSimferopol"] = "europe/simferopol";
  Timezone2["EuropeSkopje"] = "europe/skopje";
  Timezone2["EuropeSofia"] = "europe/sofia";
  Timezone2["EuropeStockholm"] = "europe/stockholm";
  Timezone2["EuropeTallinn"] = "europe/tallinn";
  Timezone2["EuropeTirane"] = "europe/tirane";
  Timezone2["EuropeUlyanovsk"] = "europe/ulyanovsk";
  Timezone2["EuropeVaduz"] = "europe/vaduz";
  Timezone2["EuropeVatican"] = "europe/vatican";
  Timezone2["EuropeVienna"] = "europe/vienna";
  Timezone2["EuropeVilnius"] = "europe/vilnius";
  Timezone2["EuropeVolgograd"] = "europe/volgograd";
  Timezone2["EuropeWarsaw"] = "europe/warsaw";
  Timezone2["EuropeZagreb"] = "europe/zagreb";
  Timezone2["EuropeZurich"] = "europe/zurich";
  Timezone2["IndianAntananarivo"] = "indian/antananarivo";
  Timezone2["IndianChagos"] = "indian/chagos";
  Timezone2["IndianChristmas"] = "indian/christmas";
  Timezone2["IndianCocos"] = "indian/cocos";
  Timezone2["IndianComoro"] = "indian/comoro";
  Timezone2["IndianKerguelen"] = "indian/kerguelen";
  Timezone2["IndianMahe"] = "indian/mahe";
  Timezone2["IndianMaldives"] = "indian/maldives";
  Timezone2["IndianMauritius"] = "indian/mauritius";
  Timezone2["IndianMayotte"] = "indian/mayotte";
  Timezone2["IndianReunion"] = "indian/reunion";
  Timezone2["PacificApia"] = "pacific/apia";
  Timezone2["PacificAuckland"] = "pacific/auckland";
  Timezone2["PacificBougainville"] = "pacific/bougainville";
  Timezone2["PacificChatham"] = "pacific/chatham";
  Timezone2["PacificChuuk"] = "pacific/chuuk";
  Timezone2["PacificEaster"] = "pacific/easter";
  Timezone2["PacificEfate"] = "pacific/efate";
  Timezone2["PacificFakaofo"] = "pacific/fakaofo";
  Timezone2["PacificFiji"] = "pacific/fiji";
  Timezone2["PacificFunafuti"] = "pacific/funafuti";
  Timezone2["PacificGalapagos"] = "pacific/galapagos";
  Timezone2["PacificGambier"] = "pacific/gambier";
  Timezone2["PacificGuadalcanal"] = "pacific/guadalcanal";
  Timezone2["PacificGuam"] = "pacific/guam";
  Timezone2["PacificHonolulu"] = "pacific/honolulu";
  Timezone2["PacificKanton"] = "pacific/kanton";
  Timezone2["PacificKiritimati"] = "pacific/kiritimati";
  Timezone2["PacificKosrae"] = "pacific/kosrae";
  Timezone2["PacificKwajalein"] = "pacific/kwajalein";
  Timezone2["PacificMajuro"] = "pacific/majuro";
  Timezone2["PacificMarquesas"] = "pacific/marquesas";
  Timezone2["PacificMidway"] = "pacific/midway";
  Timezone2["PacificNauru"] = "pacific/nauru";
  Timezone2["PacificNiue"] = "pacific/niue";
  Timezone2["PacificNorfolk"] = "pacific/norfolk";
  Timezone2["PacificNoumea"] = "pacific/noumea";
  Timezone2["PacificPagoPago"] = "pacific/pago_pago";
  Timezone2["PacificPalau"] = "pacific/palau";
  Timezone2["PacificPitcairn"] = "pacific/pitcairn";
  Timezone2["PacificPohnpei"] = "pacific/pohnpei";
  Timezone2["PacificPortMoresby"] = "pacific/port_moresby";
  Timezone2["PacificRarotonga"] = "pacific/rarotonga";
  Timezone2["PacificSaipan"] = "pacific/saipan";
  Timezone2["PacificTahiti"] = "pacific/tahiti";
  Timezone2["PacificTarawa"] = "pacific/tarawa";
  Timezone2["PacificTongatapu"] = "pacific/tongatapu";
  Timezone2["PacificWake"] = "pacific/wake";
  Timezone2["PacificWallis"] = "pacific/wallis";
  Timezone2["Utc"] = "utc";
  return Timezone2;
})(Timezone || {});

exports.Timezone = Timezone;
