// node-appwrite/dist/services/activities.mjs
import { AppwriteException } from '../client.mjs';

// src/services/activities.ts
var Activities = class {
  constructor(client) {
    this.client = client;
  }
  listEvents(paramsOrFirst) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        queries: paramsOrFirst
      };
    }
    const queries = params.queries;
    const apiPath = "/activities/events";
    const payload = {};
    if (typeof queries !== "undefined") {
      payload["queries"] = queries;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  getEvent(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        eventId: paramsOrFirst
      };
    }
    const eventId = params.eventId;
    if (typeof eventId === "undefined") {
      throw new AppwriteException('Missing required parameter: "eventId"');
    }
    const apiPath = "/activities/events/{eventId}".replace("{eventId}", encodeURIComponent(String(eventId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
};

export { Activities };
