// node-appwrite/dist/services/advisor.mjs
import { AppwriteException } from '../client.mjs';

// src/services/advisor.ts
var Advisor = class {
  constructor(client) {
    this.client = client;
  }
  listReports(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        queries: paramsOrFirst,
        total: rest[0]
      };
    }
    const queries = params.queries;
    const total = params.total;
    const apiPath = "/reports";
    const payload = {};
    if (typeof queries !== "undefined") {
      payload["queries"] = queries;
    }
    if (typeof total !== "undefined") {
      payload["total"] = total;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  getReport(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        reportId: paramsOrFirst
      };
    }
    const reportId = params.reportId;
    if (typeof reportId === "undefined") {
      throw new AppwriteException('Missing required parameter: "reportId"');
    }
    const apiPath = "/reports/{reportId}".replace("{reportId}", encodeURIComponent(String(reportId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  deleteReport(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        reportId: paramsOrFirst
      };
    }
    const reportId = params.reportId;
    if (typeof reportId === "undefined") {
      throw new AppwriteException('Missing required parameter: "reportId"');
    }
    const apiPath = "/reports/{reportId}".replace("{reportId}", encodeURIComponent(String(reportId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json"
    };
    return this.client.call(
      "delete",
      uri,
      apiHeaders,
      payload
    );
  }
  listInsights(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        reportId: paramsOrFirst,
        queries: rest[0],
        total: rest[1]
      };
    }
    const reportId = params.reportId;
    const queries = params.queries;
    const total = params.total;
    if (typeof reportId === "undefined") {
      throw new AppwriteException('Missing required parameter: "reportId"');
    }
    const apiPath = "/reports/{reportId}/insights".replace("{reportId}", encodeURIComponent(String(reportId)));
    const payload = {};
    if (typeof queries !== "undefined") {
      payload["queries"] = queries;
    }
    if (typeof total !== "undefined") {
      payload["total"] = total;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  getInsight(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        reportId: paramsOrFirst,
        insightId: rest[0]
      };
    }
    const reportId = params.reportId;
    const insightId = params.insightId;
    if (typeof reportId === "undefined") {
      throw new AppwriteException('Missing required parameter: "reportId"');
    }
    if (typeof insightId === "undefined") {
      throw new AppwriteException('Missing required parameter: "insightId"');
    }
    const apiPath = "/reports/{reportId}/insights/{insightId}".replace("{reportId}", encodeURIComponent(String(reportId))).replace("{insightId}", encodeURIComponent(String(insightId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
};

export { Advisor };
