// node-appwrite/dist/services/backups.js
'use strict';

var client = require('../client');

class Backups {
  constructor(client) {
    this.client = client;
  }
  listArchives(paramsOrFirst) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        queries: paramsOrFirst
      };
    }
    const queries = params.queries;
    const apiPath = "/backups/archives";
    const payload = {};
    if (typeof queries !== "undefined") {
      payload["queries"] = queries;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  createArchive(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst) && ("services" in paramsOrFirst || "resourceId" in paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        services: paramsOrFirst,
        resourceId: rest[0]
      };
    }
    const services = params.services;
    const resourceId = params.resourceId;
    if (typeof services === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "services"');
    }
    const apiPath = "/backups/archives";
    const payload = {};
    if (typeof services !== "undefined") {
      payload["services"] = services;
    }
    if (typeof resourceId !== "undefined") {
      payload["resourceId"] = resourceId;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  getArchive(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        archiveId: paramsOrFirst
      };
    }
    const archiveId = params.archiveId;
    if (typeof archiveId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "archiveId"');
    }
    const apiPath = "/backups/archives/{archiveId}".replace("{archiveId}", encodeURIComponent(String(archiveId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  deleteArchive(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        archiveId: paramsOrFirst
      };
    }
    const archiveId = params.archiveId;
    if (typeof archiveId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "archiveId"');
    }
    const apiPath = "/backups/archives/{archiveId}".replace("{archiveId}", encodeURIComponent(String(archiveId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "delete",
      uri,
      apiHeaders,
      payload
    );
  }
  listPolicies(paramsOrFirst) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        queries: paramsOrFirst
      };
    }
    const queries = params.queries;
    const apiPath = "/backups/policies";
    const payload = {};
    if (typeof queries !== "undefined") {
      payload["queries"] = queries;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  createPolicy(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        policyId: paramsOrFirst,
        services: rest[0],
        retention: rest[1],
        schedule: rest[2],
        name: rest[3],
        resourceId: rest[4],
        enabled: rest[5]
      };
    }
    const policyId = params.policyId;
    const services = params.services;
    const retention = params.retention;
    const schedule = params.schedule;
    const name = params.name;
    const resourceId = params.resourceId;
    const enabled = params.enabled;
    if (typeof policyId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "policyId"');
    }
    if (typeof services === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "services"');
    }
    if (typeof retention === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "retention"');
    }
    if (typeof schedule === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "schedule"');
    }
    const apiPath = "/backups/policies";
    const payload = {};
    if (typeof policyId !== "undefined") {
      payload["policyId"] = policyId;
    }
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof services !== "undefined") {
      payload["services"] = services;
    }
    if (typeof resourceId !== "undefined") {
      payload["resourceId"] = resourceId;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    if (typeof retention !== "undefined") {
      payload["retention"] = retention;
    }
    if (typeof schedule !== "undefined") {
      payload["schedule"] = schedule;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  getPolicy(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        policyId: paramsOrFirst
      };
    }
    const policyId = params.policyId;
    if (typeof policyId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "policyId"');
    }
    const apiPath = "/backups/policies/{policyId}".replace("{policyId}", encodeURIComponent(String(policyId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  updatePolicy(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        policyId: paramsOrFirst,
        name: rest[0],
        retention: rest[1],
        schedule: rest[2],
        enabled: rest[3]
      };
    }
    const policyId = params.policyId;
    const name = params.name;
    const retention = params.retention;
    const schedule = params.schedule;
    const enabled = params.enabled;
    if (typeof policyId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "policyId"');
    }
    const apiPath = "/backups/policies/{policyId}".replace("{policyId}", encodeURIComponent(String(policyId)));
    const payload = {};
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof retention !== "undefined") {
      payload["retention"] = retention;
    }
    if (typeof schedule !== "undefined") {
      payload["schedule"] = schedule;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  deletePolicy(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        policyId: paramsOrFirst
      };
    }
    const policyId = params.policyId;
    if (typeof policyId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "policyId"');
    }
    const apiPath = "/backups/policies/{policyId}".replace("{policyId}", encodeURIComponent(String(policyId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "delete",
      uri,
      apiHeaders,
      payload
    );
  }
  createRestoration(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        archiveId: paramsOrFirst,
        services: rest[0],
        newResourceId: rest[1],
        newResourceName: rest[2],
        newSpecification: rest[3]
      };
    }
    const archiveId = params.archiveId;
    const services = params.services;
    const newResourceId = params.newResourceId;
    const newResourceName = params.newResourceName;
    const newSpecification = params.newSpecification;
    if (typeof archiveId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "archiveId"');
    }
    if (typeof services === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "services"');
    }
    const apiPath = "/backups/restoration";
    const payload = {};
    if (typeof archiveId !== "undefined") {
      payload["archiveId"] = archiveId;
    }
    if (typeof services !== "undefined") {
      payload["services"] = services;
    }
    if (typeof newResourceId !== "undefined") {
      payload["newResourceId"] = newResourceId;
    }
    if (typeof newResourceName !== "undefined") {
      payload["newResourceName"] = newResourceName;
    }
    if (typeof newSpecification !== "undefined") {
      payload["newSpecification"] = newSpecification;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  listRestorations(paramsOrFirst) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        queries: paramsOrFirst
      };
    }
    const queries = params.queries;
    const apiPath = "/backups/restorations";
    const payload = {};
    if (typeof queries !== "undefined") {
      payload["queries"] = queries;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  getRestoration(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        restorationId: paramsOrFirst
      };
    }
    const restorationId = params.restorationId;
    if (typeof restorationId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "restorationId"');
    }
    const apiPath = "/backups/restorations/{restorationId}".replace("{restorationId}", encodeURIComponent(String(restorationId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
}

exports.Backups = Backups;
