// node-appwrite/dist/services/organization.mjs
import { AppwriteException } from '../client.mjs';

// src/services/organization.ts
var Organization = class {
  constructor(client) {
    this.client = client;
  }
  /**
   * Get the current organization.
   *
   * @throws {AppwriteException}
   * @returns {Promise<Models.Organization<Preferences>>}
   */
  get() {
    const apiPath = "/organization";
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  update(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        name: paramsOrFirst
      };
    }
    const name = params.name;
    if (typeof name === "undefined") {
      throw new AppwriteException('Missing required parameter: "name"');
    }
    const apiPath = "/organization";
    const payload = {};
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "put",
      uri,
      apiHeaders,
      payload
    );
  }
  /**
   * Delete the current organization. All projects that belong to the organization are deleted as well.
   *
   * @throws {AppwriteException}
   * @returns {Promise<{}>}
   */
  delete() {
    const apiPath = "/organization";
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json"
    };
    return this.client.call(
      "delete",
      uri,
      apiHeaders,
      payload
    );
  }
  listKeys(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        queries: paramsOrFirst,
        total: rest[0]
      };
    }
    const queries = params.queries;
    const total = params.total;
    const apiPath = "/organization/keys";
    const payload = {};
    if (typeof queries !== "undefined") {
      payload["queries"] = queries;
    }
    if (typeof total !== "undefined") {
      payload["total"] = total;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  createKey(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        keyId: paramsOrFirst,
        name: rest[0],
        scopes: rest[1],
        expire: rest[2]
      };
    }
    const keyId = params.keyId;
    const name = params.name;
    const scopes = params.scopes;
    const expire = params.expire;
    if (typeof keyId === "undefined") {
      throw new AppwriteException('Missing required parameter: "keyId"');
    }
    if (typeof name === "undefined") {
      throw new AppwriteException('Missing required parameter: "name"');
    }
    if (typeof scopes === "undefined") {
      throw new AppwriteException('Missing required parameter: "scopes"');
    }
    const apiPath = "/organization/keys";
    const payload = {};
    if (typeof keyId !== "undefined") {
      payload["keyId"] = keyId;
    }
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof scopes !== "undefined") {
      payload["scopes"] = scopes;
    }
    if (typeof expire !== "undefined") {
      payload["expire"] = expire;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  getKey(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        keyId: paramsOrFirst
      };
    }
    const keyId = params.keyId;
    if (typeof keyId === "undefined") {
      throw new AppwriteException('Missing required parameter: "keyId"');
    }
    const apiPath = "/organization/keys/{keyId}".replace("{keyId}", encodeURIComponent(String(keyId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  updateKey(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        keyId: paramsOrFirst,
        name: rest[0],
        scopes: rest[1],
        expire: rest[2]
      };
    }
    const keyId = params.keyId;
    const name = params.name;
    const scopes = params.scopes;
    const expire = params.expire;
    if (typeof keyId === "undefined") {
      throw new AppwriteException('Missing required parameter: "keyId"');
    }
    if (typeof name === "undefined") {
      throw new AppwriteException('Missing required parameter: "name"');
    }
    if (typeof scopes === "undefined") {
      throw new AppwriteException('Missing required parameter: "scopes"');
    }
    const apiPath = "/organization/keys/{keyId}".replace("{keyId}", encodeURIComponent(String(keyId)));
    const payload = {};
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof scopes !== "undefined") {
      payload["scopes"] = scopes;
    }
    if (typeof expire !== "undefined") {
      payload["expire"] = expire;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "put",
      uri,
      apiHeaders,
      payload
    );
  }
  deleteKey(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        keyId: paramsOrFirst
      };
    }
    const keyId = params.keyId;
    if (typeof keyId === "undefined") {
      throw new AppwriteException('Missing required parameter: "keyId"');
    }
    const apiPath = "/organization/keys/{keyId}".replace("{keyId}", encodeURIComponent(String(keyId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json"
    };
    return this.client.call(
      "delete",
      uri,
      apiHeaders,
      payload
    );
  }
  listMemberships(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        queries: paramsOrFirst,
        search: rest[0],
        total: rest[1]
      };
    }
    const queries = params.queries;
    const search = params.search;
    const total = params.total;
    const apiPath = "/organization/memberships";
    const payload = {};
    if (typeof queries !== "undefined") {
      payload["queries"] = queries;
    }
    if (typeof search !== "undefined") {
      payload["search"] = search;
    }
    if (typeof total !== "undefined") {
      payload["total"] = total;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  createMembership(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        roles: paramsOrFirst,
        email: rest[0],
        userId: rest[1],
        phone: rest[2],
        url: rest[3],
        name: rest[4]
      };
    }
    const roles = params.roles;
    const email = params.email;
    const userId = params.userId;
    const phone = params.phone;
    const url = params.url;
    const name = params.name;
    if (typeof roles === "undefined") {
      throw new AppwriteException('Missing required parameter: "roles"');
    }
    const apiPath = "/organization/memberships";
    const payload = {};
    if (typeof email !== "undefined") {
      payload["email"] = email;
    }
    if (typeof userId !== "undefined") {
      payload["userId"] = userId;
    }
    if (typeof phone !== "undefined") {
      payload["phone"] = phone;
    }
    if (typeof roles !== "undefined") {
      payload["roles"] = roles;
    }
    if (typeof url !== "undefined") {
      payload["url"] = url;
    }
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  getMembership(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        membershipId: paramsOrFirst
      };
    }
    const membershipId = params.membershipId;
    if (typeof membershipId === "undefined") {
      throw new AppwriteException('Missing required parameter: "membershipId"');
    }
    const apiPath = "/organization/memberships/{membershipId}".replace("{membershipId}", encodeURIComponent(String(membershipId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  updateMembership(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        membershipId: paramsOrFirst,
        roles: rest[0]
      };
    }
    const membershipId = params.membershipId;
    const roles = params.roles;
    if (typeof membershipId === "undefined") {
      throw new AppwriteException('Missing required parameter: "membershipId"');
    }
    if (typeof roles === "undefined") {
      throw new AppwriteException('Missing required parameter: "roles"');
    }
    const apiPath = "/organization/memberships/{membershipId}".replace("{membershipId}", encodeURIComponent(String(membershipId)));
    const payload = {};
    if (typeof roles !== "undefined") {
      payload["roles"] = roles;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  deleteMembership(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        membershipId: paramsOrFirst
      };
    }
    const membershipId = params.membershipId;
    if (typeof membershipId === "undefined") {
      throw new AppwriteException('Missing required parameter: "membershipId"');
    }
    const apiPath = "/organization/memberships/{membershipId}".replace("{membershipId}", encodeURIComponent(String(membershipId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json"
    };
    return this.client.call(
      "delete",
      uri,
      apiHeaders,
      payload
    );
  }
  listProjects(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        queries: paramsOrFirst,
        search: rest[0],
        total: rest[1]
      };
    }
    const queries = params.queries;
    const search = params.search;
    const total = params.total;
    const apiPath = "/organization/projects";
    const payload = {};
    if (typeof queries !== "undefined") {
      payload["queries"] = queries;
    }
    if (typeof search !== "undefined") {
      payload["search"] = search;
    }
    if (typeof total !== "undefined") {
      payload["total"] = total;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  createProject(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        projectId: paramsOrFirst,
        name: rest[0],
        region: rest[1]
      };
    }
    const projectId = params.projectId;
    const name = params.name;
    const region = params.region;
    if (typeof projectId === "undefined") {
      throw new AppwriteException('Missing required parameter: "projectId"');
    }
    if (typeof name === "undefined") {
      throw new AppwriteException('Missing required parameter: "name"');
    }
    const apiPath = "/organization/projects";
    const payload = {};
    if (typeof projectId !== "undefined") {
      payload["projectId"] = projectId;
    }
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof region !== "undefined") {
      payload["region"] = region;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  getProject(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        projectId: paramsOrFirst
      };
    }
    const projectId = params.projectId;
    if (typeof projectId === "undefined") {
      throw new AppwriteException('Missing required parameter: "projectId"');
    }
    const apiPath = "/organization/projects/{projectId}".replace("{projectId}", encodeURIComponent(String(projectId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  updateProject(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        projectId: paramsOrFirst,
        name: rest[0]
      };
    }
    const projectId = params.projectId;
    const name = params.name;
    if (typeof projectId === "undefined") {
      throw new AppwriteException('Missing required parameter: "projectId"');
    }
    if (typeof name === "undefined") {
      throw new AppwriteException('Missing required parameter: "name"');
    }
    const apiPath = "/organization/projects/{projectId}".replace("{projectId}", encodeURIComponent(String(projectId)));
    const payload = {};
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  deleteProject(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        projectId: paramsOrFirst
      };
    }
    const projectId = params.projectId;
    if (typeof projectId === "undefined") {
      throw new AppwriteException('Missing required parameter: "projectId"');
    }
    const apiPath = "/organization/projects/{projectId}".replace("{projectId}", encodeURIComponent(String(projectId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json"
    };
    return this.client.call(
      "delete",
      uri,
      apiHeaders,
      payload
    );
  }
};

export { Organization };
