// node-appwrite/dist/services/presences.js
'use strict';

var client = require('../client');

class Presences {
  constructor(client) {
    this.client = client;
  }
  list(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        queries: paramsOrFirst,
        total: rest[0],
        ttl: rest[1]
      };
    }
    const queries = params.queries;
    const total = params.total;
    const ttl = params.ttl;
    const apiPath = "/presences";
    const payload = {};
    if (typeof queries !== "undefined") {
      payload["queries"] = queries;
    }
    if (typeof total !== "undefined") {
      payload["total"] = total;
    }
    if (typeof ttl !== "undefined") {
      payload["ttl"] = ttl;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  get(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        presenceId: paramsOrFirst
      };
    }
    const presenceId = params.presenceId;
    if (typeof presenceId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "presenceId"');
    }
    const apiPath = "/presences/{presenceId}".replace("{presenceId}", encodeURIComponent(String(presenceId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  upsert(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        presenceId: paramsOrFirst,
        userId: rest[0],
        status: rest[1],
        permissions: rest[2],
        expiresAt: rest[3],
        metadata: rest[4]
      };
    }
    const presenceId = params.presenceId;
    const userId = params.userId;
    const status = params.status;
    const permissions = params.permissions;
    const expiresAt = params.expiresAt;
    const metadata = params.metadata;
    if (typeof presenceId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "presenceId"');
    }
    if (typeof userId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "userId"');
    }
    if (typeof status === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "status"');
    }
    const apiPath = "/presences/{presenceId}".replace("{presenceId}", encodeURIComponent(String(presenceId)));
    const payload = {};
    if (typeof userId !== "undefined") {
      payload["userId"] = userId;
    }
    if (typeof status !== "undefined") {
      payload["status"] = status;
    }
    if (typeof permissions !== "undefined") {
      payload["permissions"] = permissions;
    }
    if (typeof expiresAt !== "undefined") {
      payload["expiresAt"] = expiresAt;
    }
    if (typeof metadata !== "undefined") {
      payload["metadata"] = metadata;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "put",
      uri,
      apiHeaders,
      payload
    );
  }
  update(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        presenceId: paramsOrFirst,
        userId: rest[0],
        status: rest[1],
        expiresAt: rest[2],
        metadata: rest[3],
        permissions: rest[4],
        purge: rest[5]
      };
    }
    const presenceId = params.presenceId;
    const userId = params.userId;
    const status = params.status;
    const expiresAt = params.expiresAt;
    const metadata = params.metadata;
    const permissions = params.permissions;
    const purge = params.purge;
    if (typeof presenceId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "presenceId"');
    }
    if (typeof userId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "userId"');
    }
    const apiPath = "/presences/{presenceId}".replace("{presenceId}", encodeURIComponent(String(presenceId)));
    const payload = {};
    if (typeof userId !== "undefined") {
      payload["userId"] = userId;
    }
    if (typeof status !== "undefined") {
      payload["status"] = status;
    }
    if (typeof expiresAt !== "undefined") {
      payload["expiresAt"] = expiresAt;
    }
    if (typeof metadata !== "undefined") {
      payload["metadata"] = metadata;
    }
    if (typeof permissions !== "undefined") {
      payload["permissions"] = permissions;
    }
    if (typeof purge !== "undefined") {
      payload["purge"] = purge;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  delete(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        presenceId: paramsOrFirst
      };
    }
    const presenceId = params.presenceId;
    if (typeof presenceId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "presenceId"');
    }
    const apiPath = "/presences/{presenceId}".replace("{presenceId}", encodeURIComponent(String(presenceId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json"
    };
    return this.client.call(
      "delete",
      uri,
      apiHeaders,
      payload
    );
  }
}

exports.Presences = Presences;
