// node-appwrite/dist/services/project.js
'use strict';

var client = require('../client');

class Project {
  constructor(client) {
    this.client = client;
  }
  /**
   * Get a project.
   *
   * @throws {AppwriteException}
   * @returns {Promise<Models.Project>}
   */
  get() {
    const apiPath = "/project";
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  /**
   * Delete a project.
   *
   * @throws {AppwriteException}
   * @returns {Promise<{}>}
   */
  delete() {
    const apiPath = "/project";
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json"
    };
    return this.client.call(
      "delete",
      uri,
      apiHeaders,
      payload
    );
  }
  updateAuthMethod(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst) && ("methodId" in paramsOrFirst || "enabled" in paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        methodId: paramsOrFirst,
        enabled: rest[0]
      };
    }
    const methodId = params.methodId;
    const enabled = params.enabled;
    if (typeof methodId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "methodId"');
    }
    if (typeof enabled === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "enabled"');
    }
    const apiPath = "/project/auth-methods/{methodId}".replace("{methodId}", encodeURIComponent(String(methodId)));
    const payload = {};
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  listKeys(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        queries: paramsOrFirst,
        total: rest[0]
      };
    }
    const queries = params.queries;
    const total = params.total;
    const apiPath = "/project/keys";
    const payload = {};
    if (typeof queries !== "undefined") {
      payload["queries"] = queries;
    }
    if (typeof total !== "undefined") {
      payload["total"] = total;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  createKey(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        keyId: paramsOrFirst,
        name: rest[0],
        scopes: rest[1],
        expire: rest[2]
      };
    }
    const keyId = params.keyId;
    const name = params.name;
    const scopes = params.scopes;
    const expire = params.expire;
    if (typeof keyId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "keyId"');
    }
    if (typeof name === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "name"');
    }
    if (typeof scopes === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "scopes"');
    }
    const apiPath = "/project/keys";
    const payload = {};
    if (typeof keyId !== "undefined") {
      payload["keyId"] = keyId;
    }
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof scopes !== "undefined") {
      payload["scopes"] = scopes;
    }
    if (typeof expire !== "undefined") {
      payload["expire"] = expire;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  createEphemeralKey(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst) && ("scopes" in paramsOrFirst || "duration" in paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        scopes: paramsOrFirst,
        duration: rest[0]
      };
    }
    const scopes = params.scopes;
    const duration = params.duration;
    if (typeof scopes === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "scopes"');
    }
    if (typeof duration === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "duration"');
    }
    const apiPath = "/project/keys/ephemeral";
    const payload = {};
    if (typeof scopes !== "undefined") {
      payload["scopes"] = scopes;
    }
    if (typeof duration !== "undefined") {
      payload["duration"] = duration;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  getKey(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        keyId: paramsOrFirst
      };
    }
    const keyId = params.keyId;
    if (typeof keyId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "keyId"');
    }
    const apiPath = "/project/keys/{keyId}".replace("{keyId}", encodeURIComponent(String(keyId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  updateKey(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        keyId: paramsOrFirst,
        name: rest[0],
        scopes: rest[1],
        expire: rest[2]
      };
    }
    const keyId = params.keyId;
    const name = params.name;
    const scopes = params.scopes;
    const expire = params.expire;
    if (typeof keyId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "keyId"');
    }
    if (typeof name === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "name"');
    }
    if (typeof scopes === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "scopes"');
    }
    const apiPath = "/project/keys/{keyId}".replace("{keyId}", encodeURIComponent(String(keyId)));
    const payload = {};
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof scopes !== "undefined") {
      payload["scopes"] = scopes;
    }
    if (typeof expire !== "undefined") {
      payload["expire"] = expire;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "put",
      uri,
      apiHeaders,
      payload
    );
  }
  deleteKey(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        keyId: paramsOrFirst
      };
    }
    const keyId = params.keyId;
    if (typeof keyId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "keyId"');
    }
    const apiPath = "/project/keys/{keyId}".replace("{keyId}", encodeURIComponent(String(keyId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json"
    };
    return this.client.call(
      "delete",
      uri,
      apiHeaders,
      payload
    );
  }
  updateLabels(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        labels: paramsOrFirst
      };
    }
    const labels = params.labels;
    if (typeof labels === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "labels"');
    }
    const apiPath = "/project/labels";
    const payload = {};
    if (typeof labels !== "undefined") {
      payload["labels"] = labels;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "put",
      uri,
      apiHeaders,
      payload
    );
  }
  listMockPhones(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        queries: paramsOrFirst,
        total: rest[0]
      };
    }
    const queries = params.queries;
    const total = params.total;
    const apiPath = "/project/mock-phones";
    const payload = {};
    if (typeof queries !== "undefined") {
      payload["queries"] = queries;
    }
    if (typeof total !== "undefined") {
      payload["total"] = total;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  createMockPhone(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        number: paramsOrFirst,
        otp: rest[0]
      };
    }
    const number = params.number;
    const otp = params.otp;
    if (typeof number === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "number"');
    }
    if (typeof otp === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "otp"');
    }
    const apiPath = "/project/mock-phones";
    const payload = {};
    if (typeof number !== "undefined") {
      payload["number"] = number;
    }
    if (typeof otp !== "undefined") {
      payload["otp"] = otp;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  getMockPhone(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        number: paramsOrFirst
      };
    }
    const number = params.number;
    if (typeof number === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "number"');
    }
    const apiPath = "/project/mock-phones/{number}".replace("{number}", encodeURIComponent(String(number)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  updateMockPhone(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        number: paramsOrFirst,
        otp: rest[0]
      };
    }
    const number = params.number;
    const otp = params.otp;
    if (typeof number === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "number"');
    }
    if (typeof otp === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "otp"');
    }
    const apiPath = "/project/mock-phones/{number}".replace("{number}", encodeURIComponent(String(number)));
    const payload = {};
    if (typeof otp !== "undefined") {
      payload["otp"] = otp;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "put",
      uri,
      apiHeaders,
      payload
    );
  }
  deleteMockPhone(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        number: paramsOrFirst
      };
    }
    const number = params.number;
    if (typeof number === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "number"');
    }
    const apiPath = "/project/mock-phones/{number}".replace("{number}", encodeURIComponent(String(number)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json"
    };
    return this.client.call(
      "delete",
      uri,
      apiHeaders,
      payload
    );
  }
  listOAuth2Providers(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        queries: paramsOrFirst,
        total: rest[0]
      };
    }
    const queries = params.queries;
    const total = params.total;
    const apiPath = "/project/oauth2";
    const payload = {};
    if (typeof queries !== "undefined") {
      payload["queries"] = queries;
    }
    if (typeof total !== "undefined") {
      payload["total"] = total;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Server(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        enabled: paramsOrFirst,
        authorizationUrl: rest[0],
        scopes: rest[1],
        authorizationDetailsTypes: rest[2],
        accessTokenDuration: rest[3],
        refreshTokenDuration: rest[4],
        publicAccessTokenDuration: rest[5],
        publicRefreshTokenDuration: rest[6],
        confidentialPkce: rest[7],
        verificationUrl: rest[8],
        userCodeLength: rest[9],
        userCodeFormat: rest[10],
        deviceCodeDuration: rest[11],
        defaultScopes: rest[12]
      };
    }
    const enabled = params.enabled;
    const authorizationUrl = params.authorizationUrl;
    const scopes = params.scopes;
    const authorizationDetailsTypes = params.authorizationDetailsTypes;
    const accessTokenDuration = params.accessTokenDuration;
    const refreshTokenDuration = params.refreshTokenDuration;
    const publicAccessTokenDuration = params.publicAccessTokenDuration;
    const publicRefreshTokenDuration = params.publicRefreshTokenDuration;
    const confidentialPkce = params.confidentialPkce;
    const verificationUrl = params.verificationUrl;
    const userCodeLength = params.userCodeLength;
    const userCodeFormat = params.userCodeFormat;
    const deviceCodeDuration = params.deviceCodeDuration;
    const defaultScopes = params.defaultScopes;
    if (typeof enabled === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "enabled"');
    }
    if (typeof authorizationUrl === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "authorizationUrl"');
    }
    const apiPath = "/project/oauth2-server";
    const payload = {};
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    if (typeof authorizationUrl !== "undefined") {
      payload["authorizationUrl"] = authorizationUrl;
    }
    if (typeof scopes !== "undefined") {
      payload["scopes"] = scopes;
    }
    if (typeof authorizationDetailsTypes !== "undefined") {
      payload["authorizationDetailsTypes"] = authorizationDetailsTypes;
    }
    if (typeof accessTokenDuration !== "undefined") {
      payload["accessTokenDuration"] = accessTokenDuration;
    }
    if (typeof refreshTokenDuration !== "undefined") {
      payload["refreshTokenDuration"] = refreshTokenDuration;
    }
    if (typeof publicAccessTokenDuration !== "undefined") {
      payload["publicAccessTokenDuration"] = publicAccessTokenDuration;
    }
    if (typeof publicRefreshTokenDuration !== "undefined") {
      payload["publicRefreshTokenDuration"] = publicRefreshTokenDuration;
    }
    if (typeof confidentialPkce !== "undefined") {
      payload["confidentialPkce"] = confidentialPkce;
    }
    if (typeof verificationUrl !== "undefined") {
      payload["verificationUrl"] = verificationUrl;
    }
    if (typeof userCodeLength !== "undefined") {
      payload["userCodeLength"] = userCodeLength;
    }
    if (typeof userCodeFormat !== "undefined") {
      payload["userCodeFormat"] = userCodeFormat;
    }
    if (typeof deviceCodeDuration !== "undefined") {
      payload["deviceCodeDuration"] = deviceCodeDuration;
    }
    if (typeof defaultScopes !== "undefined") {
      payload["defaultScopes"] = defaultScopes;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "put",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Amazon(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        enabled: rest[1]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/amazon";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Apple(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        serviceId: paramsOrFirst,
        keyId: rest[0],
        teamId: rest[1],
        p8File: rest[2],
        enabled: rest[3]
      };
    }
    const serviceId = params.serviceId;
    const keyId = params.keyId;
    const teamId = params.teamId;
    const p8File = params.p8File;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/apple";
    const payload = {};
    if (typeof serviceId !== "undefined") {
      payload["serviceId"] = serviceId;
    }
    if (typeof keyId !== "undefined") {
      payload["keyId"] = keyId;
    }
    if (typeof teamId !== "undefined") {
      payload["teamId"] = teamId;
    }
    if (typeof p8File !== "undefined") {
      payload["p8File"] = p8File;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Appwrite(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        enabled: rest[1]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/appwrite";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Auth0(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        endpoint: rest[1],
        enabled: rest[2]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const endpoint = params.endpoint;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/auth0";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof endpoint !== "undefined") {
      payload["endpoint"] = endpoint;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Authentik(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        endpoint: rest[1],
        enabled: rest[2]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const endpoint = params.endpoint;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/authentik";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof endpoint !== "undefined") {
      payload["endpoint"] = endpoint;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Autodesk(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        enabled: rest[1]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/autodesk";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Bitbucket(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        key: paramsOrFirst,
        secret: rest[0],
        enabled: rest[1]
      };
    }
    const key = params.key;
    const secret = params.secret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/bitbucket";
    const payload = {};
    if (typeof key !== "undefined") {
      payload["key"] = key;
    }
    if (typeof secret !== "undefined") {
      payload["secret"] = secret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Bitly(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        enabled: rest[1]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/bitly";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Box(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        enabled: rest[1]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/box";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Dailymotion(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        apiKey: paramsOrFirst,
        apiSecret: rest[0],
        enabled: rest[1]
      };
    }
    const apiKey = params.apiKey;
    const apiSecret = params.apiSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/dailymotion";
    const payload = {};
    if (typeof apiKey !== "undefined") {
      payload["apiKey"] = apiKey;
    }
    if (typeof apiSecret !== "undefined") {
      payload["apiSecret"] = apiSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Discord(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        enabled: rest[1]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/discord";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Disqus(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        publicKey: paramsOrFirst,
        secretKey: rest[0],
        enabled: rest[1]
      };
    }
    const publicKey = params.publicKey;
    const secretKey = params.secretKey;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/disqus";
    const payload = {};
    if (typeof publicKey !== "undefined") {
      payload["publicKey"] = publicKey;
    }
    if (typeof secretKey !== "undefined") {
      payload["secretKey"] = secretKey;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Dropbox(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        appKey: paramsOrFirst,
        appSecret: rest[0],
        enabled: rest[1]
      };
    }
    const appKey = params.appKey;
    const appSecret = params.appSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/dropbox";
    const payload = {};
    if (typeof appKey !== "undefined") {
      payload["appKey"] = appKey;
    }
    if (typeof appSecret !== "undefined") {
      payload["appSecret"] = appSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Etsy(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        keyString: paramsOrFirst,
        sharedSecret: rest[0],
        enabled: rest[1]
      };
    }
    const keyString = params.keyString;
    const sharedSecret = params.sharedSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/etsy";
    const payload = {};
    if (typeof keyString !== "undefined") {
      payload["keyString"] = keyString;
    }
    if (typeof sharedSecret !== "undefined") {
      payload["sharedSecret"] = sharedSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Facebook(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        appId: paramsOrFirst,
        appSecret: rest[0],
        enabled: rest[1]
      };
    }
    const appId = params.appId;
    const appSecret = params.appSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/facebook";
    const payload = {};
    if (typeof appId !== "undefined") {
      payload["appId"] = appId;
    }
    if (typeof appSecret !== "undefined") {
      payload["appSecret"] = appSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Figma(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        enabled: rest[1]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/figma";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2FusionAuth(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        endpoint: rest[1],
        enabled: rest[2]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const endpoint = params.endpoint;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/fusionauth";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof endpoint !== "undefined") {
      payload["endpoint"] = endpoint;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2GitHub(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        enabled: rest[1]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/github";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Gitlab(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        applicationId: paramsOrFirst,
        secret: rest[0],
        endpoint: rest[1],
        enabled: rest[2]
      };
    }
    const applicationId = params.applicationId;
    const secret = params.secret;
    const endpoint = params.endpoint;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/gitlab";
    const payload = {};
    if (typeof applicationId !== "undefined") {
      payload["applicationId"] = applicationId;
    }
    if (typeof secret !== "undefined") {
      payload["secret"] = secret;
    }
    if (typeof endpoint !== "undefined") {
      payload["endpoint"] = endpoint;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Google(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        prompt: rest[1],
        enabled: rest[2]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const prompt = params.prompt;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/google";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof prompt !== "undefined") {
      payload["prompt"] = prompt;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Keycloak(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        endpoint: rest[1],
        realmName: rest[2],
        enabled: rest[3]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const endpoint = params.endpoint;
    const realmName = params.realmName;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/keycloak";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof endpoint !== "undefined") {
      payload["endpoint"] = endpoint;
    }
    if (typeof realmName !== "undefined") {
      payload["realmName"] = realmName;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Kick(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        enabled: rest[1]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/kick";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Linkedin(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        primaryClientSecret: rest[0],
        enabled: rest[1]
      };
    }
    const clientId = params.clientId;
    const primaryClientSecret = params.primaryClientSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/linkedin";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof primaryClientSecret !== "undefined") {
      payload["primaryClientSecret"] = primaryClientSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Microsoft(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        applicationId: paramsOrFirst,
        applicationSecret: rest[0],
        tenant: rest[1],
        enabled: rest[2]
      };
    }
    const applicationId = params.applicationId;
    const applicationSecret = params.applicationSecret;
    const tenant = params.tenant;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/microsoft";
    const payload = {};
    if (typeof applicationId !== "undefined") {
      payload["applicationId"] = applicationId;
    }
    if (typeof applicationSecret !== "undefined") {
      payload["applicationSecret"] = applicationSecret;
    }
    if (typeof tenant !== "undefined") {
      payload["tenant"] = tenant;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Notion(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        oauthClientId: paramsOrFirst,
        oauthClientSecret: rest[0],
        enabled: rest[1]
      };
    }
    const oauthClientId = params.oauthClientId;
    const oauthClientSecret = params.oauthClientSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/notion";
    const payload = {};
    if (typeof oauthClientId !== "undefined") {
      payload["oauthClientId"] = oauthClientId;
    }
    if (typeof oauthClientSecret !== "undefined") {
      payload["oauthClientSecret"] = oauthClientSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Oidc(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        wellKnownURL: rest[1],
        authorizationURL: rest[2],
        tokenURL: rest[3],
        userInfoURL: rest[4],
        prompt: rest[5],
        maxAge: rest[6],
        enabled: rest[7]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const wellKnownURL = params.wellKnownURL;
    const authorizationURL = params.authorizationURL;
    const tokenURL = params.tokenURL;
    const userInfoURL = params.userInfoURL;
    const prompt = params.prompt;
    const maxAge = params.maxAge;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/oidc";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof wellKnownURL !== "undefined") {
      payload["wellKnownURL"] = wellKnownURL;
    }
    if (typeof authorizationURL !== "undefined") {
      payload["authorizationURL"] = authorizationURL;
    }
    if (typeof tokenURL !== "undefined") {
      payload["tokenURL"] = tokenURL;
    }
    if (typeof userInfoURL !== "undefined") {
      payload["userInfoURL"] = userInfoURL;
    }
    if (typeof prompt !== "undefined") {
      payload["prompt"] = prompt;
    }
    if (typeof maxAge !== "undefined") {
      payload["maxAge"] = maxAge;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Okta(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        domain: rest[1],
        authorizationServerId: rest[2],
        enabled: rest[3]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const domain = params.domain;
    const authorizationServerId = params.authorizationServerId;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/okta";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof domain !== "undefined") {
      payload["domain"] = domain;
    }
    if (typeof authorizationServerId !== "undefined") {
      payload["authorizationServerId"] = authorizationServerId;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Paypal(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        secretKey: rest[0],
        enabled: rest[1]
      };
    }
    const clientId = params.clientId;
    const secretKey = params.secretKey;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/paypal";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof secretKey !== "undefined") {
      payload["secretKey"] = secretKey;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2PaypalSandbox(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        secretKey: rest[0],
        enabled: rest[1]
      };
    }
    const clientId = params.clientId;
    const secretKey = params.secretKey;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/paypalSandbox";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof secretKey !== "undefined") {
      payload["secretKey"] = secretKey;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Podio(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        enabled: rest[1]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/podio";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Salesforce(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        customerKey: paramsOrFirst,
        customerSecret: rest[0],
        enabled: rest[1]
      };
    }
    const customerKey = params.customerKey;
    const customerSecret = params.customerSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/salesforce";
    const payload = {};
    if (typeof customerKey !== "undefined") {
      payload["customerKey"] = customerKey;
    }
    if (typeof customerSecret !== "undefined") {
      payload["customerSecret"] = customerSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Slack(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        enabled: rest[1]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/slack";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Spotify(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        enabled: rest[1]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/spotify";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Stripe(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        apiSecretKey: rest[0],
        enabled: rest[1]
      };
    }
    const clientId = params.clientId;
    const apiSecretKey = params.apiSecretKey;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/stripe";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof apiSecretKey !== "undefined") {
      payload["apiSecretKey"] = apiSecretKey;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Tradeshift(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        oauth2ClientId: paramsOrFirst,
        oauth2ClientSecret: rest[0],
        enabled: rest[1]
      };
    }
    const oauth2ClientId = params.oauth2ClientId;
    const oauth2ClientSecret = params.oauth2ClientSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/tradeshift";
    const payload = {};
    if (typeof oauth2ClientId !== "undefined") {
      payload["oauth2ClientId"] = oauth2ClientId;
    }
    if (typeof oauth2ClientSecret !== "undefined") {
      payload["oauth2ClientSecret"] = oauth2ClientSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2TradeshiftSandbox(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        oauth2ClientId: paramsOrFirst,
        oauth2ClientSecret: rest[0],
        enabled: rest[1]
      };
    }
    const oauth2ClientId = params.oauth2ClientId;
    const oauth2ClientSecret = params.oauth2ClientSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/tradeshiftBox";
    const payload = {};
    if (typeof oauth2ClientId !== "undefined") {
      payload["oauth2ClientId"] = oauth2ClientId;
    }
    if (typeof oauth2ClientSecret !== "undefined") {
      payload["oauth2ClientSecret"] = oauth2ClientSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Twitch(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        enabled: rest[1]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/twitch";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2WordPress(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        enabled: rest[1]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/wordpress";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2X(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        customerKey: paramsOrFirst,
        secretKey: rest[0],
        enabled: rest[1]
      };
    }
    const customerKey = params.customerKey;
    const secretKey = params.secretKey;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/x";
    const payload = {};
    if (typeof customerKey !== "undefined") {
      payload["customerKey"] = customerKey;
    }
    if (typeof secretKey !== "undefined") {
      payload["secretKey"] = secretKey;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Yahoo(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        enabled: rest[1]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/yahoo";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Yandex(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        enabled: rest[1]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/yandex";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Zoho(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        enabled: rest[1]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/zoho";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateOAuth2Zoom(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        clientId: paramsOrFirst,
        clientSecret: rest[0],
        enabled: rest[1]
      };
    }
    const clientId = params.clientId;
    const clientSecret = params.clientSecret;
    const enabled = params.enabled;
    const apiPath = "/project/oauth2/zoom";
    const payload = {};
    if (typeof clientId !== "undefined") {
      payload["clientId"] = clientId;
    }
    if (typeof clientSecret !== "undefined") {
      payload["clientSecret"] = clientSecret;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  getOAuth2Provider(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst) && "providerId" in paramsOrFirst) {
      params = paramsOrFirst || {};
    } else {
      params = {
        providerId: paramsOrFirst
      };
    }
    const providerId = params.providerId;
    if (typeof providerId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "providerId"');
    }
    const apiPath = "/project/oauth2/{providerId}".replace("{providerId}", encodeURIComponent(String(providerId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  listPlatforms(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        queries: paramsOrFirst,
        total: rest[0]
      };
    }
    const queries = params.queries;
    const total = params.total;
    const apiPath = "/project/platforms";
    const payload = {};
    if (typeof queries !== "undefined") {
      payload["queries"] = queries;
    }
    if (typeof total !== "undefined") {
      payload["total"] = total;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  createAndroidPlatform(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst,
        name: rest[0],
        applicationId: rest[1]
      };
    }
    const platformId = params.platformId;
    const name = params.name;
    const applicationId = params.applicationId;
    if (typeof platformId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "platformId"');
    }
    if (typeof name === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "name"');
    }
    if (typeof applicationId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "applicationId"');
    }
    const apiPath = "/project/platforms/android";
    const payload = {};
    if (typeof platformId !== "undefined") {
      payload["platformId"] = platformId;
    }
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof applicationId !== "undefined") {
      payload["applicationId"] = applicationId;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  updateAndroidPlatform(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst,
        name: rest[0],
        applicationId: rest[1]
      };
    }
    const platformId = params.platformId;
    const name = params.name;
    const applicationId = params.applicationId;
    if (typeof platformId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "platformId"');
    }
    if (typeof name === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "name"');
    }
    if (typeof applicationId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "applicationId"');
    }
    const apiPath = "/project/platforms/android/{platformId}".replace("{platformId}", encodeURIComponent(String(platformId)));
    const payload = {};
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof applicationId !== "undefined") {
      payload["applicationId"] = applicationId;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "put",
      uri,
      apiHeaders,
      payload
    );
  }
  createApplePlatform(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst,
        name: rest[0],
        bundleIdentifier: rest[1]
      };
    }
    const platformId = params.platformId;
    const name = params.name;
    const bundleIdentifier = params.bundleIdentifier;
    if (typeof platformId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "platformId"');
    }
    if (typeof name === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "name"');
    }
    if (typeof bundleIdentifier === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "bundleIdentifier"');
    }
    const apiPath = "/project/platforms/apple";
    const payload = {};
    if (typeof platformId !== "undefined") {
      payload["platformId"] = platformId;
    }
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof bundleIdentifier !== "undefined") {
      payload["bundleIdentifier"] = bundleIdentifier;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  updateApplePlatform(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst,
        name: rest[0],
        bundleIdentifier: rest[1]
      };
    }
    const platformId = params.platformId;
    const name = params.name;
    const bundleIdentifier = params.bundleIdentifier;
    if (typeof platformId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "platformId"');
    }
    if (typeof name === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "name"');
    }
    if (typeof bundleIdentifier === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "bundleIdentifier"');
    }
    const apiPath = "/project/platforms/apple/{platformId}".replace("{platformId}", encodeURIComponent(String(platformId)));
    const payload = {};
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof bundleIdentifier !== "undefined") {
      payload["bundleIdentifier"] = bundleIdentifier;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "put",
      uri,
      apiHeaders,
      payload
    );
  }
  createLinuxPlatform(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst,
        name: rest[0],
        packageName: rest[1]
      };
    }
    const platformId = params.platformId;
    const name = params.name;
    const packageName = params.packageName;
    if (typeof platformId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "platformId"');
    }
    if (typeof name === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "name"');
    }
    if (typeof packageName === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "packageName"');
    }
    const apiPath = "/project/platforms/linux";
    const payload = {};
    if (typeof platformId !== "undefined") {
      payload["platformId"] = platformId;
    }
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof packageName !== "undefined") {
      payload["packageName"] = packageName;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  updateLinuxPlatform(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst,
        name: rest[0],
        packageName: rest[1]
      };
    }
    const platformId = params.platformId;
    const name = params.name;
    const packageName = params.packageName;
    if (typeof platformId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "platformId"');
    }
    if (typeof name === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "name"');
    }
    if (typeof packageName === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "packageName"');
    }
    const apiPath = "/project/platforms/linux/{platformId}".replace("{platformId}", encodeURIComponent(String(platformId)));
    const payload = {};
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof packageName !== "undefined") {
      payload["packageName"] = packageName;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "put",
      uri,
      apiHeaders,
      payload
    );
  }
  createWebPlatform(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst,
        name: rest[0],
        hostname: rest[1]
      };
    }
    const platformId = params.platformId;
    const name = params.name;
    const hostname = params.hostname;
    if (typeof platformId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "platformId"');
    }
    if (typeof name === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "name"');
    }
    if (typeof hostname === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "hostname"');
    }
    const apiPath = "/project/platforms/web";
    const payload = {};
    if (typeof platformId !== "undefined") {
      payload["platformId"] = platformId;
    }
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof hostname !== "undefined") {
      payload["hostname"] = hostname;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  updateWebPlatform(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst,
        name: rest[0],
        hostname: rest[1]
      };
    }
    const platformId = params.platformId;
    const name = params.name;
    const hostname = params.hostname;
    if (typeof platformId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "platformId"');
    }
    if (typeof name === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "name"');
    }
    if (typeof hostname === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "hostname"');
    }
    const apiPath = "/project/platforms/web/{platformId}".replace("{platformId}", encodeURIComponent(String(platformId)));
    const payload = {};
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof hostname !== "undefined") {
      payload["hostname"] = hostname;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "put",
      uri,
      apiHeaders,
      payload
    );
  }
  createWindowsPlatform(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst,
        name: rest[0],
        packageIdentifierName: rest[1]
      };
    }
    const platformId = params.platformId;
    const name = params.name;
    const packageIdentifierName = params.packageIdentifierName;
    if (typeof platformId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "platformId"');
    }
    if (typeof name === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "name"');
    }
    if (typeof packageIdentifierName === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "packageIdentifierName"');
    }
    const apiPath = "/project/platforms/windows";
    const payload = {};
    if (typeof platformId !== "undefined") {
      payload["platformId"] = platformId;
    }
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof packageIdentifierName !== "undefined") {
      payload["packageIdentifierName"] = packageIdentifierName;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  updateWindowsPlatform(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst,
        name: rest[0],
        packageIdentifierName: rest[1]
      };
    }
    const platformId = params.platformId;
    const name = params.name;
    const packageIdentifierName = params.packageIdentifierName;
    if (typeof platformId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "platformId"');
    }
    if (typeof name === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "name"');
    }
    if (typeof packageIdentifierName === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "packageIdentifierName"');
    }
    const apiPath = "/project/platforms/windows/{platformId}".replace("{platformId}", encodeURIComponent(String(platformId)));
    const payload = {};
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof packageIdentifierName !== "undefined") {
      payload["packageIdentifierName"] = packageIdentifierName;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "put",
      uri,
      apiHeaders,
      payload
    );
  }
  getPlatform(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst
      };
    }
    const platformId = params.platformId;
    if (typeof platformId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "platformId"');
    }
    const apiPath = "/project/platforms/{platformId}".replace("{platformId}", encodeURIComponent(String(platformId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  deletePlatform(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        platformId: paramsOrFirst
      };
    }
    const platformId = params.platformId;
    if (typeof platformId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "platformId"');
    }
    const apiPath = "/project/platforms/{platformId}".replace("{platformId}", encodeURIComponent(String(platformId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json"
    };
    return this.client.call(
      "delete",
      uri,
      apiHeaders,
      payload
    );
  }
  listPolicies(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        queries: paramsOrFirst,
        total: rest[0]
      };
    }
    const queries = params.queries;
    const total = params.total;
    const apiPath = "/project/policies";
    const payload = {};
    if (typeof queries !== "undefined") {
      payload["queries"] = queries;
    }
    if (typeof total !== "undefined") {
      payload["total"] = total;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  updateDenyAliasedEmailPolicy(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        enabled: paramsOrFirst
      };
    }
    const enabled = params.enabled;
    if (typeof enabled === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "enabled"');
    }
    const apiPath = "/project/policies/deny-aliased-email";
    const payload = {};
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateDenyCorporateEmailPolicy(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        enabled: paramsOrFirst
      };
    }
    const enabled = params.enabled;
    if (typeof enabled === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "enabled"');
    }
    const apiPath = "/project/policies/deny-corporate-email";
    const payload = {};
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateDenyDisposableEmailPolicy(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        enabled: paramsOrFirst
      };
    }
    const enabled = params.enabled;
    if (typeof enabled === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "enabled"');
    }
    const apiPath = "/project/policies/deny-disposable-email";
    const payload = {};
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateDenyFreeEmailPolicy(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        enabled: paramsOrFirst
      };
    }
    const enabled = params.enabled;
    if (typeof enabled === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "enabled"');
    }
    const apiPath = "/project/policies/deny-free-email";
    const payload = {};
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateMembershipPrivacyPolicy(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        userId: paramsOrFirst,
        userEmail: rest[0],
        userPhone: rest[1],
        userName: rest[2],
        userMFA: rest[3],
        userAccessedAt: rest[4]
      };
    }
    const userId = params.userId;
    const userEmail = params.userEmail;
    const userPhone = params.userPhone;
    const userName = params.userName;
    const userMFA = params.userMFA;
    const userAccessedAt = params.userAccessedAt;
    const apiPath = "/project/policies/membership-privacy";
    const payload = {};
    if (typeof userId !== "undefined") {
      payload["userId"] = userId;
    }
    if (typeof userEmail !== "undefined") {
      payload["userEmail"] = userEmail;
    }
    if (typeof userPhone !== "undefined") {
      payload["userPhone"] = userPhone;
    }
    if (typeof userName !== "undefined") {
      payload["userName"] = userName;
    }
    if (typeof userMFA !== "undefined") {
      payload["userMFA"] = userMFA;
    }
    if (typeof userAccessedAt !== "undefined") {
      payload["userAccessedAt"] = userAccessedAt;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updatePasswordDictionaryPolicy(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        enabled: paramsOrFirst
      };
    }
    const enabled = params.enabled;
    if (typeof enabled === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "enabled"');
    }
    const apiPath = "/project/policies/password-dictionary";
    const payload = {};
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updatePasswordHistoryPolicy(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        total: paramsOrFirst
      };
    }
    const total = params.total;
    if (typeof total === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "total"');
    }
    const apiPath = "/project/policies/password-history";
    const payload = {};
    if (typeof total !== "undefined") {
      payload["total"] = total;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updatePasswordPersonalDataPolicy(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        enabled: paramsOrFirst
      };
    }
    const enabled = params.enabled;
    if (typeof enabled === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "enabled"');
    }
    const apiPath = "/project/policies/password-personal-data";
    const payload = {};
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updatePasswordStrengthPolicy(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        min: paramsOrFirst,
        uppercase: rest[0],
        lowercase: rest[1],
        number: rest[2],
        symbols: rest[3]
      };
    }
    const min = params.min;
    const uppercase = params.uppercase;
    const lowercase = params.lowercase;
    const number = params.number;
    const symbols = params.symbols;
    const apiPath = "/project/policies/password-strength";
    const payload = {};
    if (typeof min !== "undefined") {
      payload["min"] = min;
    }
    if (typeof uppercase !== "undefined") {
      payload["uppercase"] = uppercase;
    }
    if (typeof lowercase !== "undefined") {
      payload["lowercase"] = lowercase;
    }
    if (typeof number !== "undefined") {
      payload["number"] = number;
    }
    if (typeof symbols !== "undefined") {
      payload["symbols"] = symbols;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateSessionAlertPolicy(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        enabled: paramsOrFirst
      };
    }
    const enabled = params.enabled;
    if (typeof enabled === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "enabled"');
    }
    const apiPath = "/project/policies/session-alert";
    const payload = {};
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateSessionDurationPolicy(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        duration: paramsOrFirst
      };
    }
    const duration = params.duration;
    if (typeof duration === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "duration"');
    }
    const apiPath = "/project/policies/session-duration";
    const payload = {};
    if (typeof duration !== "undefined") {
      payload["duration"] = duration;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateSessionInvalidationPolicy(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        enabled: paramsOrFirst
      };
    }
    const enabled = params.enabled;
    if (typeof enabled === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "enabled"');
    }
    const apiPath = "/project/policies/session-invalidation";
    const payload = {};
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateSessionLimitPolicy(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        total: paramsOrFirst
      };
    }
    const total = params.total;
    if (typeof total === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "total"');
    }
    const apiPath = "/project/policies/session-limit";
    const payload = {};
    if (typeof total !== "undefined") {
      payload["total"] = total;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateUserLimitPolicy(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        total: paramsOrFirst
      };
    }
    const total = params.total;
    if (typeof total === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "total"');
    }
    const apiPath = "/project/policies/user-limit";
    const payload = {};
    if (typeof total !== "undefined") {
      payload["total"] = total;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  getPolicy(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst) && "policyId" in paramsOrFirst) {
      params = paramsOrFirst || {};
    } else {
      params = {
        policyId: paramsOrFirst
      };
    }
    const policyId = params.policyId;
    if (typeof policyId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "policyId"');
    }
    const apiPath = "/project/policies/{policyId}".replace("{policyId}", encodeURIComponent(String(policyId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  updateProtocol(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst) && ("protocolId" in paramsOrFirst || "enabled" in paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        protocolId: paramsOrFirst,
        enabled: rest[0]
      };
    }
    const protocolId = params.protocolId;
    const enabled = params.enabled;
    if (typeof protocolId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "protocolId"');
    }
    if (typeof enabled === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "enabled"');
    }
    const apiPath = "/project/protocols/{protocolId}".replace("{protocolId}", encodeURIComponent(String(protocolId)));
    const payload = {};
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateService(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst) && ("serviceId" in paramsOrFirst || "enabled" in paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        serviceId: paramsOrFirst,
        enabled: rest[0]
      };
    }
    const serviceId = params.serviceId;
    const enabled = params.enabled;
    if (typeof serviceId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "serviceId"');
    }
    if (typeof enabled === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "enabled"');
    }
    const apiPath = "/project/services/{serviceId}".replace("{serviceId}", encodeURIComponent(String(serviceId)));
    const payload = {};
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  updateSMTP(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        host: paramsOrFirst,
        port: rest[0],
        username: rest[1],
        password: rest[2],
        senderEmail: rest[3],
        senderName: rest[4],
        replyToEmail: rest[5],
        replyToName: rest[6],
        secure: rest[7],
        enabled: rest[8]
      };
    }
    const host = params.host;
    const port = params.port;
    const username = params.username;
    const password = params.password;
    const senderEmail = params.senderEmail;
    const senderName = params.senderName;
    const replyToEmail = params.replyToEmail;
    const replyToName = params.replyToName;
    const secure = params.secure;
    const enabled = params.enabled;
    const apiPath = "/project/smtp";
    const payload = {};
    if (typeof host !== "undefined") {
      payload["host"] = host;
    }
    if (typeof port !== "undefined") {
      payload["port"] = port;
    }
    if (typeof username !== "undefined") {
      payload["username"] = username;
    }
    if (typeof password !== "undefined") {
      payload["password"] = password;
    }
    if (typeof senderEmail !== "undefined") {
      payload["senderEmail"] = senderEmail;
    }
    if (typeof senderName !== "undefined") {
      payload["senderName"] = senderName;
    }
    if (typeof replyToEmail !== "undefined") {
      payload["replyToEmail"] = replyToEmail;
    }
    if (typeof replyToName !== "undefined") {
      payload["replyToName"] = replyToName;
    }
    if (typeof secure !== "undefined") {
      payload["secure"] = secure;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  createSMTPTest(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        emails: paramsOrFirst
      };
    }
    const emails = params.emails;
    if (typeof emails === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "emails"');
    }
    const apiPath = "/project/smtp/tests";
    const payload = {};
    if (typeof emails !== "undefined") {
      payload["emails"] = emails;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  listEmailTemplates(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        queries: paramsOrFirst,
        total: rest[0]
      };
    }
    const queries = params.queries;
    const total = params.total;
    const apiPath = "/project/templates/email";
    const payload = {};
    if (typeof queries !== "undefined") {
      payload["queries"] = queries;
    }
    if (typeof total !== "undefined") {
      payload["total"] = total;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  updateEmailTemplate(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst) && ("templateId" in paramsOrFirst || "locale" in paramsOrFirst || "subject" in paramsOrFirst || "message" in paramsOrFirst || "senderName" in paramsOrFirst || "senderEmail" in paramsOrFirst || "replyToEmail" in paramsOrFirst || "replyToName" in paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        templateId: paramsOrFirst,
        locale: rest[0],
        subject: rest[1],
        message: rest[2],
        senderName: rest[3],
        senderEmail: rest[4],
        replyToEmail: rest[5],
        replyToName: rest[6]
      };
    }
    const templateId = params.templateId;
    const locale = params.locale;
    const subject = params.subject;
    const message = params.message;
    const senderName = params.senderName;
    const senderEmail = params.senderEmail;
    const replyToEmail = params.replyToEmail;
    const replyToName = params.replyToName;
    if (typeof templateId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "templateId"');
    }
    const apiPath = "/project/templates/email";
    const payload = {};
    if (typeof templateId !== "undefined") {
      payload["templateId"] = templateId;
    }
    if (typeof locale !== "undefined") {
      payload["locale"] = locale;
    }
    if (typeof subject !== "undefined") {
      payload["subject"] = subject;
    }
    if (typeof message !== "undefined") {
      payload["message"] = message;
    }
    if (typeof senderName !== "undefined") {
      payload["senderName"] = senderName;
    }
    if (typeof senderEmail !== "undefined") {
      payload["senderEmail"] = senderEmail;
    }
    if (typeof replyToEmail !== "undefined") {
      payload["replyToEmail"] = replyToEmail;
    }
    if (typeof replyToName !== "undefined") {
      payload["replyToName"] = replyToName;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
  getEmailTemplate(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst) && ("templateId" in paramsOrFirst || "locale" in paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        templateId: paramsOrFirst,
        locale: rest[0]
      };
    }
    const templateId = params.templateId;
    const locale = params.locale;
    if (typeof templateId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "templateId"');
    }
    const apiPath = "/project/templates/email/{templateId}".replace("{templateId}", encodeURIComponent(String(templateId)));
    const payload = {};
    if (typeof locale !== "undefined") {
      payload["locale"] = locale;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  listVariables(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        queries: paramsOrFirst,
        total: rest[0]
      };
    }
    const queries = params.queries;
    const total = params.total;
    const apiPath = "/project/variables";
    const payload = {};
    if (typeof queries !== "undefined") {
      payload["queries"] = queries;
    }
    if (typeof total !== "undefined") {
      payload["total"] = total;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  createVariable(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        variableId: paramsOrFirst,
        key: rest[0],
        value: rest[1],
        secret: rest[2]
      };
    }
    const variableId = params.variableId;
    const key = params.key;
    const value = params.value;
    const secret = params.secret;
    if (typeof variableId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "variableId"');
    }
    if (typeof key === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "key"');
    }
    if (typeof value === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "value"');
    }
    const apiPath = "/project/variables";
    const payload = {};
    if (typeof variableId !== "undefined") {
      payload["variableId"] = variableId;
    }
    if (typeof key !== "undefined") {
      payload["key"] = key;
    }
    if (typeof value !== "undefined") {
      payload["value"] = value;
    }
    if (typeof secret !== "undefined") {
      payload["secret"] = secret;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  getVariable(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        variableId: paramsOrFirst
      };
    }
    const variableId = params.variableId;
    if (typeof variableId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "variableId"');
    }
    const apiPath = "/project/variables/{variableId}".replace("{variableId}", encodeURIComponent(String(variableId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  updateVariable(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        variableId: paramsOrFirst,
        key: rest[0],
        value: rest[1],
        secret: rest[2]
      };
    }
    const variableId = params.variableId;
    const key = params.key;
    const value = params.value;
    const secret = params.secret;
    if (typeof variableId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "variableId"');
    }
    const apiPath = "/project/variables/{variableId}".replace("{variableId}", encodeURIComponent(String(variableId)));
    const payload = {};
    if (typeof key !== "undefined") {
      payload["key"] = key;
    }
    if (typeof value !== "undefined") {
      payload["value"] = value;
    }
    if (typeof secret !== "undefined") {
      payload["secret"] = secret;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "put",
      uri,
      apiHeaders,
      payload
    );
  }
  deleteVariable(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        variableId: paramsOrFirst
      };
    }
    const variableId = params.variableId;
    if (typeof variableId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "variableId"');
    }
    const apiPath = "/project/variables/{variableId}".replace("{variableId}", encodeURIComponent(String(variableId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json"
    };
    return this.client.call(
      "delete",
      uri,
      apiHeaders,
      payload
    );
  }
}

exports.Project = Project;
