// node-appwrite/dist/services/proxy.mjs
import { AppwriteException } from '../client.mjs';

// src/services/proxy.ts
var Proxy = class {
  constructor(client) {
    this.client = client;
  }
  listRules(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        queries: paramsOrFirst,
        total: rest[0]
      };
    }
    const queries = params.queries;
    const total = params.total;
    const apiPath = "/proxy/rules";
    const payload = {};
    if (typeof queries !== "undefined") {
      payload["queries"] = queries;
    }
    if (typeof total !== "undefined") {
      payload["total"] = total;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  createAPIRule(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        domain: paramsOrFirst
      };
    }
    const domain = params.domain;
    if (typeof domain === "undefined") {
      throw new AppwriteException('Missing required parameter: "domain"');
    }
    const apiPath = "/proxy/rules/api";
    const payload = {};
    if (typeof domain !== "undefined") {
      payload["domain"] = domain;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  createFunctionRule(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        domain: paramsOrFirst,
        functionId: rest[0],
        branch: rest[1]
      };
    }
    const domain = params.domain;
    const functionId = params.functionId;
    const branch = params.branch;
    if (typeof domain === "undefined") {
      throw new AppwriteException('Missing required parameter: "domain"');
    }
    if (typeof functionId === "undefined") {
      throw new AppwriteException('Missing required parameter: "functionId"');
    }
    const apiPath = "/proxy/rules/function";
    const payload = {};
    if (typeof domain !== "undefined") {
      payload["domain"] = domain;
    }
    if (typeof functionId !== "undefined") {
      payload["functionId"] = functionId;
    }
    if (typeof branch !== "undefined") {
      payload["branch"] = branch;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  createRedirectRule(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        domain: paramsOrFirst,
        url: rest[0],
        statusCode: rest[1],
        resourceId: rest[2],
        resourceType: rest[3]
      };
    }
    const domain = params.domain;
    const url = params.url;
    const statusCode = params.statusCode;
    const resourceId = params.resourceId;
    const resourceType = params.resourceType;
    if (typeof domain === "undefined") {
      throw new AppwriteException('Missing required parameter: "domain"');
    }
    if (typeof url === "undefined") {
      throw new AppwriteException('Missing required parameter: "url"');
    }
    if (typeof statusCode === "undefined") {
      throw new AppwriteException('Missing required parameter: "statusCode"');
    }
    if (typeof resourceId === "undefined") {
      throw new AppwriteException('Missing required parameter: "resourceId"');
    }
    if (typeof resourceType === "undefined") {
      throw new AppwriteException('Missing required parameter: "resourceType"');
    }
    const apiPath = "/proxy/rules/redirect";
    const payload = {};
    if (typeof domain !== "undefined") {
      payload["domain"] = domain;
    }
    if (typeof url !== "undefined") {
      payload["url"] = url;
    }
    if (typeof statusCode !== "undefined") {
      payload["statusCode"] = statusCode;
    }
    if (typeof resourceId !== "undefined") {
      payload["resourceId"] = resourceId;
    }
    if (typeof resourceType !== "undefined") {
      payload["resourceType"] = resourceType;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  createSiteRule(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        domain: paramsOrFirst,
        siteId: rest[0],
        branch: rest[1]
      };
    }
    const domain = params.domain;
    const siteId = params.siteId;
    const branch = params.branch;
    if (typeof domain === "undefined") {
      throw new AppwriteException('Missing required parameter: "domain"');
    }
    if (typeof siteId === "undefined") {
      throw new AppwriteException('Missing required parameter: "siteId"');
    }
    const apiPath = "/proxy/rules/site";
    const payload = {};
    if (typeof domain !== "undefined") {
      payload["domain"] = domain;
    }
    if (typeof siteId !== "undefined") {
      payload["siteId"] = siteId;
    }
    if (typeof branch !== "undefined") {
      payload["branch"] = branch;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  getRule(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        ruleId: paramsOrFirst
      };
    }
    const ruleId = params.ruleId;
    if (typeof ruleId === "undefined") {
      throw new AppwriteException('Missing required parameter: "ruleId"');
    }
    const apiPath = "/proxy/rules/{ruleId}".replace("{ruleId}", encodeURIComponent(String(ruleId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  deleteRule(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        ruleId: paramsOrFirst
      };
    }
    const ruleId = params.ruleId;
    if (typeof ruleId === "undefined") {
      throw new AppwriteException('Missing required parameter: "ruleId"');
    }
    const apiPath = "/proxy/rules/{ruleId}".replace("{ruleId}", encodeURIComponent(String(ruleId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json"
    };
    return this.client.call(
      "delete",
      uri,
      apiHeaders,
      payload
    );
  }
  updateRuleStatus(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        ruleId: paramsOrFirst
      };
    }
    const ruleId = params.ruleId;
    if (typeof ruleId === "undefined") {
      throw new AppwriteException('Missing required parameter: "ruleId"');
    }
    const apiPath = "/proxy/rules/{ruleId}/status".replace("{ruleId}", encodeURIComponent(String(ruleId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
};

export { Proxy };
