// node-appwrite/dist/services/webhooks.mjs
import { AppwriteException } from '../client.mjs';

// src/services/webhooks.ts
var Webhooks = class {
  constructor(client) {
    this.client = client;
  }
  list(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        queries: paramsOrFirst,
        total: rest[0]
      };
    }
    const queries = params.queries;
    const total = params.total;
    const apiPath = "/webhooks";
    const payload = {};
    if (typeof queries !== "undefined") {
      payload["queries"] = queries;
    }
    if (typeof total !== "undefined") {
      payload["total"] = total;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  create(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        webhookId: paramsOrFirst,
        url: rest[0],
        name: rest[1],
        events: rest[2],
        enabled: rest[3],
        tls: rest[4],
        authUsername: rest[5],
        authPassword: rest[6],
        secret: rest[7]
      };
    }
    const webhookId = params.webhookId;
    const url = params.url;
    const name = params.name;
    const events = params.events;
    const enabled = params.enabled;
    const tls = params.tls;
    const authUsername = params.authUsername;
    const authPassword = params.authPassword;
    const secret = params.secret;
    if (typeof webhookId === "undefined") {
      throw new AppwriteException('Missing required parameter: "webhookId"');
    }
    if (typeof url === "undefined") {
      throw new AppwriteException('Missing required parameter: "url"');
    }
    if (typeof name === "undefined") {
      throw new AppwriteException('Missing required parameter: "name"');
    }
    if (typeof events === "undefined") {
      throw new AppwriteException('Missing required parameter: "events"');
    }
    const apiPath = "/webhooks";
    const payload = {};
    if (typeof webhookId !== "undefined") {
      payload["webhookId"] = webhookId;
    }
    if (typeof url !== "undefined") {
      payload["url"] = url;
    }
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof events !== "undefined") {
      payload["events"] = events;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    if (typeof tls !== "undefined") {
      payload["tls"] = tls;
    }
    if (typeof authUsername !== "undefined") {
      payload["authUsername"] = authUsername;
    }
    if (typeof authPassword !== "undefined") {
      payload["authPassword"] = authPassword;
    }
    if (typeof secret !== "undefined") {
      payload["secret"] = secret;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  get(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        webhookId: paramsOrFirst
      };
    }
    const webhookId = params.webhookId;
    if (typeof webhookId === "undefined") {
      throw new AppwriteException('Missing required parameter: "webhookId"');
    }
    const apiPath = "/webhooks/{webhookId}".replace("{webhookId}", encodeURIComponent(String(webhookId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "accept": "application/json"
    };
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  update(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        webhookId: paramsOrFirst,
        name: rest[0],
        url: rest[1],
        events: rest[2],
        enabled: rest[3],
        tls: rest[4],
        authUsername: rest[5],
        authPassword: rest[6]
      };
    }
    const webhookId = params.webhookId;
    const name = params.name;
    const url = params.url;
    const events = params.events;
    const enabled = params.enabled;
    const tls = params.tls;
    const authUsername = params.authUsername;
    const authPassword = params.authPassword;
    if (typeof webhookId === "undefined") {
      throw new AppwriteException('Missing required parameter: "webhookId"');
    }
    if (typeof name === "undefined") {
      throw new AppwriteException('Missing required parameter: "name"');
    }
    if (typeof url === "undefined") {
      throw new AppwriteException('Missing required parameter: "url"');
    }
    if (typeof events === "undefined") {
      throw new AppwriteException('Missing required parameter: "events"');
    }
    const apiPath = "/webhooks/{webhookId}".replace("{webhookId}", encodeURIComponent(String(webhookId)));
    const payload = {};
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof url !== "undefined") {
      payload["url"] = url;
    }
    if (typeof events !== "undefined") {
      payload["events"] = events;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    if (typeof tls !== "undefined") {
      payload["tls"] = tls;
    }
    if (typeof authUsername !== "undefined") {
      payload["authUsername"] = authUsername;
    }
    if (typeof authPassword !== "undefined") {
      payload["authPassword"] = authPassword;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "put",
      uri,
      apiHeaders,
      payload
    );
  }
  delete(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        webhookId: paramsOrFirst
      };
    }
    const webhookId = params.webhookId;
    if (typeof webhookId === "undefined") {
      throw new AppwriteException('Missing required parameter: "webhookId"');
    }
    const apiPath = "/webhooks/{webhookId}".replace("{webhookId}", encodeURIComponent(String(webhookId)));
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json"
    };
    return this.client.call(
      "delete",
      uri,
      apiHeaders,
      payload
    );
  }
  updateSecret(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        webhookId: paramsOrFirst,
        secret: rest[0]
      };
    }
    const webhookId = params.webhookId;
    const secret = params.secret;
    if (typeof webhookId === "undefined") {
      throw new AppwriteException('Missing required parameter: "webhookId"');
    }
    const apiPath = "/webhooks/{webhookId}/secret".replace("{webhookId}", encodeURIComponent(String(webhookId)));
    const payload = {};
    if (typeof secret !== "undefined") {
      payload["secret"] = secret;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "X-Appwrite-Project": this.client.config.project,
      "content-type": "application/json",
      "accept": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
};

export { Webhooks };
